{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd #for handling csv and csv contents\n",
    "from rdflib import Graph, Literal, RDF, URIRef, Namespace #basic RDF handling\n",
    "from rdflib.namespace import FOAF , XSD, RDFS #most common namespaces\n",
    "import urllib.parse #for parsing strings to URI's"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = Graph()\n",
    "title = Namespace('http://example.org/title/')\n",
    "author = Namespace('http://example.org/author/')\n",
    "year = Namespace('http://example.org/year/')\n",
    "doi = Namespace('http://example.org/doi/')\n",
    "pmid = Namespace('http://example.org/PMID/')\n",
    "journal = Namespace('http://example.org/journal/')\n",
    "volume = Namespace('http://example.org/volume/')\n",
    "abstract = Namespace('http://example.org/abstract/')\n",
    "keyword_abstract = Namespace('http://example.org/keyword_abstract/')\n",
    "keyword_mesh = Namespace('http://example.org/keywords_mesh/')\n",
    "keyword_merged = Namespace('http://example.org/keywords_merged/')\n",
    "links = Namespace('http://example.org/fulltext_links/')\n",
    "articles = Namespace('http://example.org/related_articles/')\n",
    "fields = Namespace('http://example.org/fields/')\n",
    "subfields = Namespace('http://example.org/subfield/')\n",
    "aimethod = Namespace('http://example.org/ai_method')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_json(\"extractUntil2020.json\")\n",
    "df.to_csv()\n",
    "print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for index, row in df.iterrows():\n",
    "    g.add((URIRef(pmid+str(row['PMID'])),RDF.type, Literal(row['PMID'])))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(title+'Title'), Literal(row['PMID']))) \n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(author+'name'), Literal(row['Authors'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(doi+str(row['PMID'])), URIRef(doi+'doi'),Literal(row['Doi'], datatype=XSD.integer) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(journal+'journal'), Literal(row['Journal'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(volume+'volume'), Literal(row['Volume'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(abstract+'abstract'), Literal(row['Abstract'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(keyword_abstract+'key_abs'), Literal(row['Keywords_Abstract'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(keyword_mesh+'key_mesh'), Literal(row['Keywords_MeSH'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), URIRef(keyword_merged+'key_merged'), Literal(row['Keywords_Merged'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), RDFS.comment, Literal(row['FullTextLinks'], datatype=XSD.string) ))\n",
    "    g.add((URIRef(pmid+str(row['PMID'])), RDFS.seeAlso, Literal(row['Related Articles'], datatype=XSD.string) ))\n",
    "    #g.add((URIRef(pmid+row['PMID']), URIRef(field+'field'), Literal(row['Field'], datatype=XSD.string) ))\n",
    "    #g.add((URIRef(pmid+row['PMID']), RDFS.subclassOf, Literal(row['Subfield'], datatype=XSD.string) ))\n",
    "    #g.add((URIRef(pmid+row['PMID']), URIRef(aimethod+'aimethod'), Literal(row['ai_method'])))\n",
    "    #g.add((URIRef(loc+urllib.parse.quote(row['Address'])), URIRef(schema+'name'), Literal(row['Address'], datatype=XSD.string) ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@prefix ns1: <http://example.org/keywords_merged/> .\n",
      "@prefix ns2: <http://example.org/title/> .\n",
      "@prefix ns3: <http://example.org/doi/> .\n",
      "@prefix ns4: <http://example.org/volume/> .\n",
      "@prefix ns5: <http://example.org/journal/> .\n",
      "@prefix ns6: <http://example.org/keywords_mesh/> .\n",
      "@prefix ns7: <http://example.org/keyword_abstract/> .\n",
      "@prefix ns8: <http://example.org/author/> .\n",
      "@prefix ns9: <http://example.org/abstract/> .\n",
      "@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .\n",
      "@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\n",
      "\n",
      "<http://example.org/PMID/24694170> a 24694170 ;\n",
      "    ns9:abstract \" The automatic identification of epileptic EEG signals is significant in both relieving heavy workload of visual inspection of EEG recordings and treatment of epilepsy. This paper presents a novel method based on the theory of sparse representation to identify epileptic EEGs. At first, the raw EEG epochs are preprocessed via Gaussian low pass filtering and differential operation. Then, in the scheme of sparse representation based classification (SRC), a test EEG sample is sparsely represented on the training set by solving l1-minimization problem, and the represented residuals associated with ictal and interictal training samples are computed. The test EEG sample is categorized as the class that yields the minimum represented residual. So unlike the conventional EEG classification methods, the choice and calculation of EEG features are avoided in the proposed framework. Moreover, the kernel trick is employed to generate a kernel version of the SRC method for improving the separability between ictal and interictal classes. The satisfactory recognition accuracy of 98.63% for ictal and interictal EEG classification and for ictal and normal EEG classification has been achieved by the kernel SRC. In addition, the fast speed makes the kernel SRC suit for the real-time seizure monitoring application in the near future. \"^^xsd:string ;\n",
      "    ns8:name \"Qi Yuan; Weidong Zhou; Shasha Yuan; Xueli Li; Jiwen Wang; Guijuan Jia; \"^^xsd:string ;\n",
      "    ns5:journal \"Int J Neural Syst\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Artificial Intelligence*;  Electroencephalography / classification*;  Electroencephalography / methods*;  Epilepsy / diagnosis*;  Humans;  Signal Processing, Computer-Assisted*;  Support Vector Machine;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Artificial Intelligence*;  Electroencephalography / classification*;  Electroencephalography / methods*;  Epilepsy / diagnosis*;  Humans;  Signal Processing, Computer-Assisted*;  Support Vector Machine;  \"^^xsd:string ;\n",
      "    ns2:Title 24694170 ;\n",
      "    ns4:volume \"24\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.worldscientific.com/doi/10.1142/S0129065714500154?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub  0pubmed']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30076538,25653073,21616643,27458376,16013270,32655355,31865469,30603222,30510569,30076538\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/24955405> a 24955405 ;\n",
      "    ns9:abstract \" Principal component analysis or PCA has been traditionally used as one of the feature extraction techniques in face recognition systems yielding high accuracy when requiring a small number of features. However, the covariance matrix and eigenvalue decomposition stages cause high computational complexity, especially for a large database. Thus, this research presents an alternative approach utilizing an Expectation-Maximization algorithm to reduce the determinant matrix manipulation resulting in the reduction of the stages' complexity. To improve the computational time, a novel parallel architecture was employed to utilize the benefits of parallelization of matrix computation during feature extraction and classification stages including parallel preprocessing, and their combinations, so-called a Parallel Expectation-Maximization PCA architecture. Comparing to a traditional PCA and its derivatives, the results indicate lower complexity with an insignificant difference in recognition precision leading to high speed face recognition systems, that is, the speed-up over nine and three times over PCA and Parallel PCA. \"^^xsd:string ;\n",
      "    ns8:name \"Kanokmon Rujirakul; Chakchai So-In; Banchar Arnonkijpanich; \"^^xsd:string ;\n",
      "    ns5:journal \"ScientificWorldJournal\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Algorithms;  Artificial Intelligence;  Face*;  Humans;  Pattern Recognition, Automated*;  Principal Component Analysis*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Artificial Intelligence;  Face*;  Humans;  Pattern Recognition, Automated*;  Principal Component Analysis*;  \"^^xsd:string ;\n",
      "    ns2:Title 24955405 ;\n",
      "    ns4:volume \"2014\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1155/2014/468176', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/24955405/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"16724592,15460279,19789108,32656082,28050560\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/25291742> a 25291742 ;\n",
      "    ns9:abstract \" In adaptive dynamic programming, neurocontrol, and reinforcement learning, the objective is for an agent to learn to choose actions so as to minimize a total cost function. In this paper, we show that when discretized time is used to model the motion of the agent, it can be very important to do clipping on the motion of the agent in the final time step of the trajectory. By clipping, we mean that the final time step of the trajectory is to be truncated such that the agent stops exactly at the first terminal state reached, and no distance further. We demonstrate that when clipping is omitted, learning performance can fail to reach the optimum, and when clipping is done properly, learning performance can improve significantly. The clipping problem we describe affects algorithms that use explicit derivatives of the model functions of the environment to calculate a learning gradient. These include backpropagation through time for control and methods based on dual heuristic programming. However, the clipping problem does not significantly affect methods based on heuristic dynamic programming, temporal differences learning, or policy-gradient learning algorithms. \"^^xsd:string ;\n",
      "    ns8:name \"Michael Fairbank; Danil Prokhorov; Eduardo Alonso; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Neural Netw Learn Syst\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Algorithms;  Artificial Intelligence*;  Computer Simulation;  Humans;  Neural Networks, Computer*;  Nonlinear Dynamics*;  Reinforcement, Psychology*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Artificial Intelligence*;  Computer Simulation;  Humans;  Neural Networks, Computer*;  Nonlinear Dynamics*;  Reinforcement, Psychology*;  \"^^xsd:string ;\n",
      "    ns2:Title 25291742 ;\n",
      "    ns4:volume \"25\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TNNLS.2014.2297991']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"19842990,25751878,23954546,19386468,24995352\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/25570647> a 25570647 ;\n",
      "    ns9:abstract \" The cerebellum plays a critical role for sensorimotor control and learning. However dysmertria or delays in movements' onsets consequent to damages in cerebellum cannot be cured completely at the moment. To foster a potential cure based on neuroprosthetic technology, we present a frame-based Network-on-Chip (NoC) hardware architecture for implementing a bio-realistic cerebellum model with 100,000 neurons, which has been used for studying timing control or passage-of-time (POT) encoding mediated by the cerebellum. The results demonstrate that our implementation can reproduce the POT functionality properly. The computational speed can achieve to 25.6 ms for simulating 1 sec real world activities. Furthermore, we show a hardware electronic setup and illustrate how the silicon cerebellum can be adapted as a potential neuroprosthetic platform for future biological or clinical applications. \"^^xsd:string ;\n",
      "    ns8:name \"Junwen Luo; Graeme Coapes; Terrence Mak; Tadashi Yamazaki; Chung Tin; Patrick Degenaar; \"^^xsd:string ;\n",
      "    ns5:journal \"Conf Proc IEEE Eng Med Biol Soc\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Animals;  Artificial Intelligence;  Cerebellum / physiology*;  Electronics;  Models, Biological*;  Neural Networks, Computer;  Time Factors;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Animals;  Artificial Intelligence;  Cerebellum / physiology*;  Electronics;  Models, Biological*;  Neural Networks, Computer;  Time Factors;  \"^^xsd:string ;\n",
      "    ns2:Title 25570647 ;\n",
      "    ns4:volume \"2014\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/EMBC.2014.6944279']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"26452290,29115280,17953620,19495900,18807937\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/28820010> a 28820010 ;\n",
      "    ns9:abstract \" This study examines the clinical decision support systems in healthcare, in particular about the prevention, diagnosis and treatment of respiratory diseases, such as Asthma and chronic obstructive pulmonary disease. The empirical pulmonology study of a representative sample (n = 132) attempts to identify the major factors that contribute to the diagnosis of these diseases. Machine learning results show that in chronic obstructive pulmonary disease's case, Random Forest classifier outperforms other techniques with 97.7 per cent precision, while the most prominent attributes for diagnosis are smoking, forced expiratory volume 1, age and forced vital capacity. In asthma's case, the best precision, 80.3 per cent, is achieved again with the Random Forest classifier, while the most prominent attribute is MEF2575. \"^^xsd:string ;\n",
      "    ns8:name \"Dimitris Spathis; Panayiotis Vlamos; \"^^xsd:string ;\n",
      "    ns5:journal \"Health Informatics J\"^^xsd:string ;\n",
      "    ns7:key_abs \"  asthma; chronic obstructive pulmonary disease; clinical decision making; machine learning; respiratory diseases; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  asthma; chronic obstructive pulmonary disease; clinical decision making; machine learning; respiratory diseases;   Adolescent;  Adult;  Aged;  Aged, 80 and over;  Asthma / diagnosis*;  Child;  Child, Preschool;  Female;  Humans;  Infant;  Machine Learning / trends*;  Male;  Middle Aged;  Pulmonary Disease, Chronic Obstructive / diagnosis*;  Respiratory Function Tests / methods;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adolescent;  Adult;  Aged;  Aged, 80 and over;  Asthma / diagnosis*;  Child;  Child, Preschool;  Female;  Humans;  Infant;  Machine Learning / trends*;  Male;  Middle Aged;  Pulmonary Disease, Chronic Obstructive / diagnosis*;  Respiratory Function Tests / methods;  \"^^xsd:string ;\n",
      "    ns2:Title 28820010 ;\n",
      "    ns4:volume \"25\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://journals.sagepub.com/doi/10.1177/1460458217723169?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub  0pubmed']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28088797,30425470,27616884,16612962,15786604,32732930,31371934,30344655,30076356\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/29237568> a 29237568 ;\n",
      "    ns9:abstract \" The gut-brain axis communicates the brain with the gut microbiota, a bidirectional conduit that has received increasing attention in recent years thanks to its emerging role in brain development and function. Alterations in microbiota composition have been associated to neurological and psychiatric disorders, and several studies suggest that the immune system plays a fundamental role in the gut-brain interaction. Recent advances in brain imaging and in microbiome sequencing have generated a large amount of information, yet the data from both these sources need to be combined efficiently to extract biological meaning, and any diagnostic and/or prognostic benefit from these tools. In addition, the causal nature of the gut-brain interaction remains to be fully established, and preclinical findings translated to humans. In this \\\"Perspective\\\" article, we discuss recent efforts to combine data on the gut microbiota with the features that can be obtained from the conversion of brain images into mineable data. The subsequent analysis of these data for diagnostic and prognostic purposes is an approach we call radiomicrobiomics and it holds tremendous potential to enhance our understanding of this fascinating connection. \"^^xsd:string ;\n",
      "    ns8:name \"Silvia De Santis; David Moratal; Santiago Canals; \"^^xsd:string ;\n",
      "    ns5:journal \"Neuroscience\"^^xsd:string ;\n",
      "    ns7:key_abs \"  advanced MRI; big data; gut-brain axis; machine learning; microbiota; radiomicrobiomics; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  advanced MRI; big data; gut-brain axis; machine learning; microbiota; radiomicrobiomics;   Research Support, Non-U.S. Gov't;  Animals;  Big Data;  Brain / diagnostic imaging*;  Brain / growth & development;  Brain / physiology;  Computational Biology / methods*;  Gastrointestinal Microbiome* / physiology;  Humans;  Machine Learning;  Magnetic Resonance Imaging / methods;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Animals;  Big Data;  Brain / diagnostic imaging*;  Brain / growth & development;  Brain / physiology;  Computational Biology / methods*;  Gastrointestinal Microbiome* / physiology;  Humans;  Machine Learning;  Magnetic Resonance Imaging / methods;  \"^^xsd:string ;\n",
      "    ns2:Title 29237568 ;\n",
      "    ns4:volume \"403\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0306-4522(17)30876-X']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29492874,28852308,27647198,27005587,26046241,31456743\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/29546796> a 29546796 ;\n",
      "    ns9:abstract \" Nicotine dependence is the primary addictive stage of cigarette smoking. Although a lot of studies have been performed to explore the molecular mechanism underlying nicotine dependence, our understanding on this disorder is still far from complete. Over the past decades, an increasing number of candidate genes involved in nicotine dependence have been identified by different technical approaches, including the genetic association analysis. In this study, we performed a comprehensive collection of candidate genes reported to be genetically associated with nicotine dependence. Then, the biochemical pathways enriched in these genes were identified by considering the gene's propensity to be related to nicotine dependence. One of the most widely used pathway enrichment analysis approach, over-representation analysis, ignores the function non-equivalence of genes in candidate gene set and may have low discriminative power in identifying some dysfunctional pathways. To overcome such drawbacks, we constructed a comprehensive human protein-protein interaction network, and then assigned a function weighting score to each candidate gene based on their network topological features. Evaluation indicated the function weighting score scheme was consistent with available evidence. Finally, the function weighting scores of the candidate genes were incorporated into pathway analysis to identify the dysfunctional pathways involved in nicotine dependence, and the interactions between pathways was detected by pathway crosstalk analysis. Compared to conventional over-representation-based pathway analysis tool, the modified method exhibited improved discriminative power and detected some novel pathways potentially underlying nicotine dependence. In summary, we conducted a comprehensive collection of genes associated with nicotine dependence and then detected the biochemical pathways enriched in these genes using a modified pathway enrichment analysis approach with function weighting score of candidate genes integrated. Our results may provide insight into the molecular mechanism underlying nicotine dependence. \"^^xsd:string ;\n",
      "    ns8:name \"Ying Hu; Zhonghai Fang; Yichen Yang; Ting Fan; Ju Wang; \"^^xsd:string ;\n",
      "    ns5:journal \"J Biomol Struct Dyn\"^^xsd:string ;\n",
      "    ns7:key_abs \"  gene function; network; nicotine dependence; over-representation analysis; pathway; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  gene function; network; nicotine dependence; over-representation analysis; pathway;   Computational Biology / methods;  Gene Expression Profiling;  Gene Expression Regulation / drug effects*;  Gene Ontology;  Gene Regulatory Networks;  Humans;  Nicotine / pharmacology*;  Protein Interaction Mapping* / methods;  Protein Interaction Maps*;  Signal Transduction / drug effects*;  Smoking / adverse effects;  Nicotine;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Computational Biology / methods;  Gene Expression Profiling;  Gene Expression Regulation / drug effects*;  Gene Ontology;  Gene Regulatory Networks;  Humans;  Nicotine / pharmacology*;  Protein Interaction Mapping* / methods;  Protein Interaction Maps*;  Signal Transduction / drug effects*;  Smoking / adverse effects;  Nicotine;  \"^^xsd:string ;\n",
      "    ns2:Title 29546796 ;\n",
      "    ns4:volume \"37\"^^xsd:string ;\n",
      "    rdfs:comment \"['None']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28623929,30132538,25965070,29129810,32311908\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/29990046> a 29990046 ;\n",
      "    ns9:abstract \" Computational protein structure prediction is a long-standing challenge in bioinformatics. In the process of predicting protein 3D structures, it is common that parts of an experimental structure are missing or parts of a predicted structure need to be remodeled. The process of predicting local protein structures of particular regions is called loop modeling. In this paper, five new loop modeling methods based on machine learning techniques, called NearLooper, ConLooper, ResLooper, HyLooper1, and HyLooper2 are proposed. NearLooper is based on the nearest neighbor technique. ConLooper applies deep convolutional neural networks to predict ${\\\\mathrm{C}}_{{{\\\\alpha }}}$Cα atoms distance matrix as an orientation-independent representation of protein structure. ResLooper uses residual neural networks instead of deep convolutional neural networks. HyLooper1 combines the results of NearLooper and ConLooper while HyLooper2 combines NearLooper and ResLooper. Three commonly used benchmarks for loop modeling are used to compare the performance between these methods and existing state-of-the-art methods. The experiment results show promising performance in which our best method improves existing state-of-the-art methods by 28 and 54 percent of average RMSD on two datasets while being comparable on the other one. \"^^xsd:string ;\n",
      "    ns8:name \"Son P Nguyen; Zhaoyu Li; Dong Xu; Yi Shang; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE/ACM Trans Comput Biol Bioinform\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, N.I.H., Extramural;  Research Support, U.S. Gov't, Non-P.H.S.;  Algorithms;  Computational Biology / methods*;  Deep Learning*;  Proteins / chemistry*;  Proteins;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, U.S. Gov't, Non-P.H.S.;  Algorithms;  Computational Biology / methods*;  Deep Learning*;  Proteins / chemistry*;  Proteins;  \"^^xsd:string ;\n",
      "    ns2:Title 29990046 ;\n",
      "    ns4:volume \"16\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TCBB.2017.2784434', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/29990046/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31208331,30075707,30808287,29452923,29522900,30676995\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/29993442> a 29993442 ;\n",
      "    ns9:abstract \"  Objective:  Recognition of human activities and mental states using wearable sensors and smartphones has attracted considerable attention recently. In particular, prediction of the stress level of a subject using an electrocardiogram sensor has been studied extensively. In this paper, we attempt to predict the degree of concentration by using heart-rate features. However, due to strong diversity in individuals and high sampling costs, building an accurate prediction model is still highly challenging.   Method:  To overcome these difficulties, we propose to use a multitask learning (MTL) technique for effectively sharing information among similar individuals.   Result:  Through experiments with 18 healthy subjects performing daily office works, such as writing reports, we demonstrate that the proposed method significantly improves the accuracy of concentration prediction in small sample situations.   Conclusion:  The performance of the MTL method is shown to be stable across different subjects, which is an important advantage over conventional models.   Significance:  This improvement has significant impact in real-world concentration recognition because the data collection burden of each user can be drastically mitigated. \"^^xsd:string ;\n",
      "    ns8:name \"Hirotaka Kaji; Hisashi Iizuka; Masashi Sugiyama; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Biomed Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Adult;  Attention / physiology*;  Brain Waves / physiology*;  Electrocardiography / methods*;  Electroencephalography / methods*;  Equipment Design;  Female;  Human Activities;  Humans;  Machine Learning;  Male;  Middle Aged;  Signal Processing, Computer-Assisted*;  Wearable Electronic Devices;  Wireless Technology;  Young Adult;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adult;  Attention / physiology*;  Brain Waves / physiology*;  Electrocardiography / methods*;  Electroencephalography / methods*;  Equipment Design;  Female;  Human Activities;  Humans;  Machine Learning;  Male;  Middle Aged;  Signal Processing, Computer-Assisted*;  Wearable Electronic Devices;  Wireless Technology;  Young Adult;  \"^^xsd:string ;\n",
      "    ns2:Title 29993442 ;\n",
      "    ns4:volume \"66\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TBME.2018.2830366']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31623279,29758969,25073164,29993585,26272456,31739605,31003522\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/29994031> a 29994031 ;\n",
      "    ns9:abstract \" DNA methylation is a critical epigenetic modification that plays an important role in cancers. The available algorithms fail to fully characterize epigenetic modules. To address this issue, we first characterize the epigenetic module as a group of well-connected genes in the protein interaction network and are also co-methylated based on gene methylation profiles. Then, the epigenetic module discovery problem is transformed into an optimization problem. Then, a regularized nonnegative matrix factorization algorithm for methylation modules (RNMF-MM) is presented, where the co-methylation constraint is treated as a regularizer. Using the artificial networks with known module structure, we demonstrate that the proposed algorithm outperforms state-of-the-art approaches in terms of accuracy. On the basis of breast cancer methylation data and protein interaction network, the RNMF-MM algorithm discovers methylation modules that are significantly more enriched by the known pathways than those obtained by other algorithms. These modules serve as biomarkers for predicting cancer stages and estimating survival time of patients. The proposed model and algorithm provide an effective way for the integrative analysis of protein interaction network and methylation data. \"^^xsd:string ;\n",
      "    ns8:name \"Xiaoke Ma; Penggang Sun; Zhong-Yuan Zhang; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE/ACM Trans Comput Biol Bioinform\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Algorithms;  Biomarkers / chemistry;  Breast Neoplasms / chemistry;  Computational Biology / methods*;  DNA Methylation;  Epigenesis, Genetic*;  Female;  Gene Expression Regulation, Neoplastic;  Humans;  Kaplan-Meier Estimate;  Methylation;  Models, Statistical;  Neural Networks, Computer;  Protein Interaction Mapping*;  Proteins / chemistry;  Biomarkers;  Proteins;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Algorithms;  Biomarkers / chemistry;  Breast Neoplasms / chemistry;  Computational Biology / methods*;  DNA Methylation;  Epigenesis, Genetic*;  Female;  Gene Expression Regulation, Neoplastic;  Humans;  Kaplan-Meier Estimate;  Methylation;  Models, Statistical;  Neural Networks, Computer;  Protein Interaction Mapping*;  Proteins / chemistry;  Biomarkers;  Proteins;  \"^^xsd:string ;\n",
      "    ns2:Title 29994031 ;\n",
      "    ns4:volume \"16\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TCBB.2018.2831666']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28137264,29240706,24794928,29701681,26987530,32373160,32174978,31496635,31086568\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30040632> a 30040632 ;\n",
      "    ns9:abstract \" Volumetric lung tumor segmentation and accurate longitudinal tracking of tumor volume changes from computed tomography images are essential for monitoring tumor response to therapy. Hence, we developed two multiple resolution residually connected network (MRRN) formulations called incremental-MRRN and dense-MRRN. Our networks simultaneously combine features across multiple image resolution and feature levels through residual connections to detect and segment the lung tumors. We evaluated our method on a total of 1210 non-small cell (NSCLC) lung tumors and nodules from three data sets consisting of 377 tumors from the open-source Cancer Imaging Archive (TCIA), 304 advanced stage NSCLC treated with anti- PD-1 checkpoint immunotherapy from internal institution MSKCC data set, and 529 lung nodules from the Lung Image Database Consortium (LIDC). The algorithm was trained using 377 tumors from the TCIA data set and validated on the MSKCC and tested on LIDC data sets. The segmentation accuracy compared to expert delineations was evaluated by computing the dice similarity coefficient, Hausdorff distances, sensitivity, and precision metrics. Our best performing incremental-MRRN method produced the highest DSC of 0.74 ± 0.13 for TCIA, 0.75±0.12 for MSKCC, and 0.68±0.23 for the LIDC data sets. There was no significant difference in the estimations of volumetric tumor changes computed using the incremental-MRRN method compared with the expert segmentation. In summary, we have developed a multi-scale CNN approach for volumetrically segmenting lung tumors which enables accurate, automated identification of and serial measurement of tumor volumes in the lung. \"^^xsd:string ;\n",
      "    ns8:name \"Jue Jiang; Yu-Chi Hu; Chia-Ju Liu; Darragh Halpenny; Matthew D Hellmann; Joseph O Deasy; Gig Mageras; Harini Veeraraghavan; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Med Imaging\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Algorithms;  Databases, Factual;  Deep Learning;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Lung / diagnostic imaging;  Lung Neoplasms / diagnostic imaging*;  Tomography, X-Ray Computed / methods*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Algorithms;  Databases, Factual;  Deep Learning;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Lung / diagnostic imaging;  Lung Neoplasms / diagnostic imaging*;  Tomography, X-Ray Computed / methods*;  \"^^xsd:string ;\n",
      "    ns2:Title 30040632 ;\n",
      "    ns4:volume \"38\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TMI.2018.2857800', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30040632/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31274206,26055544,23020972,29450843,32030668,32773965,32395724,31841402,31354393\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30123948> a 30123948 ;\n",
      "    ns9:abstract \" Early detection of breast tumors, feet pre-ulcers diagnosing in diabetic patients, and identifying the location of pain in patients are essential to physicians. Hot or cold regions in medical thermographic images have potential to be suspicious. Hence extracting the hottest or coldest regions in the body thermographic images is an important task. Lazy snapping is an interactive image cutout algorithm that can be applied to extract the hottest or coldest regions in the body thermographic images quickly with easy detailed adjustment. The most important advantage of this technique is that it can provide the results for physicians in real time readily. In other words, it is a good interactive image segmentation algorithm since it has two basic characteristics: (1) the algorithm produces intuitive segmentation that reflects the user intent with given a certain user input and (2) the algorithm is efficient enough to provide instant visual feedback. Comparing to other methods used by the authors for segmentation of breast thermograms such as K-means, fuzzy c-means, level set, and mean shift algorithms, lazy snapping was more user-friendly and could provide instant visual feedback. In this study, twelve test cases were presented and by applying lazy snapping algorithm, the hottest or coldest regions were extracted from the corresponding body thermographic images. The time taken to see the results varied from 7 to 30 s for these twelve cases. It was concluded that lazy snapping was much faster than other methods applied by the authors such as K-means, fuzzy c-means, level set, and mean shift algorithms for segmentation. Graphical abstract Time taken to implement lazy snapping algorithm to extract suspicious regions in different presented thermograms (in seconds). In this study, ten test cases are presented that by applying lazy snapping algorithm, the hottest or coldest regions were extracted from the corresponding body thermographic images. The time taken to see the results varied from 7 to 30 s for the ten cases. It concludes lazy snapping is much faster than other methods applied by the authors. \"^^xsd:string ;\n",
      "    ns8:name \"Mahnaz Etehadtavakol; Zahra Emrani; E Y K Ng; \"^^xsd:string ;\n",
      "    ns5:journal \"Med Biol Eng Comput\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Coldest; Extracting; Hottest; Images; Lazy snapping; Thermography; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Coldest; Extracting; Hottest; Images; Lazy snapping; Thermography;   Algorithms;  Breast / pathology;  Breast Neoplasms / diagnosis*;  Female;  Fuzzy Logic;  Hot Temperature;  Humans;  Image Interpretation, Computer-Assisted / methods;  Middle Aged;  Pattern Recognition, Automated / methods;  Thermography / methods;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Breast / pathology;  Breast Neoplasms / diagnosis*;  Female;  Fuzzy Logic;  Hot Temperature;  Humans;  Image Interpretation, Computer-Assisted / methods;  Middle Aged;  Pattern Recognition, Automated / methods;  Thermography / methods;  \"^^xsd:string ;\n",
      "    ns2:Title 30123948 ;\n",
      "    ns4:volume \"57\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/s11517-018-1876-2']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"20192053,26417258,25934260,25605141,20384268,32235780,31915519\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30189255> a 30189255 ;\n",
      "    ns9:abstract \" Accurate and reliable prediction of clinical progression over time has the potential to improve the outcomes of chronic disease. The classical approach to analyzing longitudinal data is to use (generalized) linear mixed-effect models (GLMM). However, linear parametric models are predicated on assumptions, which are often difficult to verify. In contrast, data-driven machine learning methods can be applied to derive insight from the raw data without a priori assumptions. However, the underlying theory of most machine learning algorithms assume that the data is independent and identically distributed, making them inefficient for longitudinal supervised learning. In this study, we formulate an analytic framework, which integrates the random-effects structure of GLMM into non-linear machine learning models capable of exploiting temporal heterogeneous effects, sparse and varying-length patient characteristics inherent in longitudinal data. We applied the derived mixed-effect machine learning (MEml) framework to predict longitudinal change in glycemic control measured by hemoglobin A1c (HbA1c) among well controlled adults with type 2 diabetes. Results show that MEml is competitive with traditional GLMM, but substantially outperformed standard machine learning models that do not account for random-effects. Specifically, the accuracy of MEml in predicting glycemic change at the 1st, 2nd, 3rd, and 4th clinical visits in advanced was 1.04, 1.08, 1.11, and 1.14 times that of the gradient boosted model respectively, with similar results for the other methods. To further demonstrate the general applicability of MEml, a series of experiments were performed using real publicly available and synthetic data sets for accuracy and robustness. These experiments reinforced the superiority of MEml over the other methods. Overall, results from this study highlight the importance of modeling random-effects in machine learning approaches based on longitudinal data. Our MEml method is highly resistant to correlated data, readily accounts for random-effects, and predicts change of a longitudinal clinical outcome in real-world clinical settings with high accuracy. \"^^xsd:string ;\n",
      "    ns8:name \"Che Ngufor; Holly Van Houten; Brian S Caffo; Nilay D Shah; Rozalina G McCoy; \"^^xsd:string ;\n",
      "    ns5:journal \"J Biomed Inform\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Glycemic control; Glycosylated hemoglobin; Longitudinal supervised learning; Machine learning; Random-effects; Type 2 diabetes; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Glycemic control; Glycosylated hemoglobin; Longitudinal supervised learning; Machine learning; Random-effects; Type 2 diabetes;   Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Adult;  Algorithms;  Diabetes Mellitus, Type 2 / blood;  Diabetes Mellitus, Type 2 / diagnosis*;  Disease Progression;  Glycated Hemoglobin A / analysis*;  Humans;  Longitudinal Studies;  Machine Learning*;  Glycated Hemoglobin A;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Adult;  Algorithms;  Diabetes Mellitus, Type 2 / blood;  Diabetes Mellitus, Type 2 / diagnosis*;  Disease Progression;  Glycated Hemoglobin A / analysis*;  Humans;  Longitudinal Studies;  Machine Learning*;  Glycated Hemoglobin A;  \"^^xsd:string ;\n",
      "    ns2:Title 30189255 ;\n",
      "    ns4:volume \"89\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S1532-0464(18)30175-8', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30189255/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28922296,31185988,27919371,28393586,26604100,32678448,30984108\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30206809> a 30206809 ;\n",
      "    ns9:abstract \" Principal component analysis is a widely used method for the dimensionality reduction of a given data set in a high-dimensional Euclidean space. Here we define and analyze two analogues of principal component analysis in the setting of tropical geometry. In one approach, we study the Stiefel tropical linear space of fixed dimension closest to the data points in the tropical projective torus; in the other approach, we consider the tropical polytope with a fixed number of vertices closest to the data points. We then give approximative algorithms for both approaches and apply them to phylogenetics, testing the methods on simulated phylogenetic data and on an empirical dataset of Apicomplexa genomes. \"^^xsd:string ;\n",
      "    ns8:name \"Ruriko Yoshida; Leon Zhang; Xu Zhang; \"^^xsd:string ;\n",
      "    ns5:journal \"Bull Math Biol\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Dimensionality reduction; Phylogenomics; Tropical geometry; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Dimensionality reduction; Phylogenomics; Tropical geometry;   Research Support, Non-U.S. Gov't;  Research Support, U.S. Gov't, Non-P.H.S.;  Algorithms;  Apicomplexa / classification;  Apicomplexa / genetics;  Computational Biology;  Computer Heuristics;  Computer Simulation;  Databases, Genetic / statistics & numerical data;  Genome, Protozoan;  Linear Models;  Mathematical Concepts;  Models, Genetic;  Models, Statistical;  Phylogeny*;  Principal Component Analysis*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Research Support, U.S. Gov't, Non-P.H.S.;  Algorithms;  Apicomplexa / classification;  Apicomplexa / genetics;  Computational Biology;  Computer Heuristics;  Computer Simulation;  Databases, Genetic / statistics & numerical data;  Genome, Protozoan;  Linear Models;  Mathematical Concepts;  Models, Genetic;  Models, Statistical;  Phylogeny*;  Principal Component Analysis*;  \"^^xsd:string ;\n",
      "    ns2:Title 30206809 ;\n",
      "    ns4:volume \"81\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/s11538-018-0493-4']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"32516398,23424133,29422694,18814680,19635681\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30334752> a 30334752 ;\n",
      "    ns9:abstract \" Prostate cancer is the most common and second most deadly form of cancer in men in the United States. The classification of prostate cancers based on Gleason grading using histological images is important in risk assessment and treatment planning for patients. Here, we demonstrate a new region-based convolutional neural network framework for multi-task prediction using an epithelial network head and a grading network head. Compared with a single-task model, our multi-task model can provide complementary contextual information, which contributes to better performance. Our model is achieved a state-of-the-art performance in epithelial cells detection and Gleason grading tasks simultaneously. Using fivefold cross-validation, our model is achieved an epithelial cells detection accuracy of 99.07% with an average area under the curve of 0.998. As for Gleason grading, our model is obtained a mean intersection over union of 79.56% and an overall pixel accuracy of 89.40%. \"^^xsd:string ;\n",
      "    ns8:name \"Wenyuan Li; Jiayun Li; Karthik V Sarma; King Chung Ho; Shiwen Shen; Beatrice S Knudsen; Arkadiusz Gertych; Corey W Arnold; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Med Imaging\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Histocytochemistry;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Male;  Neoplasm Grading / methods*;  Neural Networks, Computer*;  Prostate / diagnostic imaging*;  Prostatic Neoplasms / diagnostic imaging*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Histocytochemistry;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Male;  Neoplasm Grading / methods*;  Neural Networks, Computer*;  Prostate / diagnostic imaging*;  Prostatic Neoplasms / diagnostic imaging*;  \"^^xsd:string ;\n",
      "    ns2:Title 30334752 ;\n",
      "    ns4:volume \"38\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TMI.2018.2875868', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30334752/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30835218,31098801,30848813,26991008,30694570,32294962,31131293,31063138\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30346279> a 30346279 ;\n",
      "    ns9:abstract \"  Objective:  Segmentation of anatomical structures in ultrasound images requires vast radiological knowledge and experience. Moreover, the manual segmentation often results in subjective variations, therefore, an automatic segmentation is desirable. We aim to develop a fully convolutional neural network (FCNN) with attentional deep supervision for the automatic and accurate segmentation of the ultrasound images.   Method:  FCNN/CNNs are used to infer high-level context using low-level image features. In this paper, a sub-problem specific deep supervision of the FCNN is performed. The attention of fine resolution layers is steered to learn object boundary definitions using auxiliary losses, whereas coarse resolution layers are trained to discriminate object regions from the background. Furthermore, a customized scheme for downweighting the auxiliary losses and a trainable fusion layer are introduced. This produces an accurate segmentation and helps in dealing with the broken boundaries, usually found in the ultrasound images.   Results:  The proposed network is first tested for blood vessel segmentation in liver images. It results in F1 score, mean intersection over union, and dice index of 0.83, 0.83, and 0.79, respectively. The best values observed among the existing approaches are produced by U-net as 0.74, 0.81, and 0.75, respectively. The proposed network also results in dice index value of 0.91 in the lumen segmentation experiments on MICCAI 2011 IVUS challenge dataset, which is near to the provided reference value of 0.93. Furthermore, the improvements similar to vessel segmentation experiments are also observed in the experiment performed to segment lesions.   Conclusion:  Deep supervision of the network based on the input-output characteristics of the layers results in improvement in overall segmentation accuracy.   Significance:  Sub-problem specific deep supervision for ultrasound image segmentation is the main contribution of this paper. Currently the network is trained and tested for fixed size inputs. It requires image resizing and limits the performance in small size images. \"^^xsd:string ;\n",
      "    ns8:name \"Deepak Mishra; Santanu Chaudhury; Mukul Sarkar; Arvinder Singh Soin; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Biomed Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Blood Vessels / diagnostic imaging;  Humans;  Image Processing, Computer-Assisted / methods*;  Liver / diagnostic imaging;  Neural Networks, Computer*;  Supervised Machine Learning*;  Ultrasonography / methods*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Blood Vessels / diagnostic imaging;  Humans;  Image Processing, Computer-Assisted / methods*;  Liver / diagnostic imaging;  Neural Networks, Computer*;  Supervised Machine Learning*;  Ultrasonography / methods*;  \"^^xsd:string ;\n",
      "    ns2:Title 30346279 ;\n",
      "    ns4:volume \"66\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TBME.2018.2877577']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30136285,31356693,31074513,32030668,28445774,31851956,31673567,31630868\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30522862> a 30522862 ;\n",
      "    ns9:abstract \" High-throughput methodologies and machine learning have been central in developing systems-level perspectives in molecular biology. Unfortunately, performing such integrative analyses has traditionally been reserved for bioinformaticians. This is now changing with the appearance of resources to help bench-side biologists become skilled at computational data analysis and handling large omics data sets. Here, we show an entry route into the field of omics data analytics. We provide information about easily accessible data sources and suggest some first steps for aspiring computational data analysts. Moreover, we highlight how machine learning is transforming the field and how it can help make sense of biological data. Finally, we suggest good starting points for self-learning and hope to convince readers that computational data analysis and programming are not intimidating. \"^^xsd:string ;\n",
      "    ns8:name \"Piotr Grabowski; Juri Rappsilber; \"^^xsd:string ;\n",
      "    ns5:journal \"Trends Biochem Sci\"^^xsd:string ;\n",
      "    ns7:key_abs \"  data integration; data science; functional genomics; machine learning; systems biology; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  data integration; data science; functional genomics; machine learning; systems biology;   Review;  Animals;  Data Science*;  Genomics*;  High-Throughput Screening Assays;  Humans;  Machine Learning*;  Software*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Review;  Animals;  Data Science*;  Genomics*;  High-Throughput Screening Assays;  Humans;  Machine Learning*;  Software*;  \"^^xsd:string ;\n",
      "    ns2:Title 30522862 ;\n",
      "    ns4:volume \"44\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0968-0004(18)30226-3', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30522862/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29699486,27474269,30267935,30696086,31349257,32180906,31818883\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30526357> a 30526357 ;\n",
      "    ns9:abstract \"\"^^xsd:string ;\n",
      "    ns8:name \"Ciprian Catana; \"^^xsd:string ;\n",
      "    ns5:journal \"Radiology\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Comment;  Aniline Compounds;  Deep Learning*;  Magnetic Resonance Imaging*;  Positron-Emission Tomography;  Stilbenes;  Aniline Compounds;  Stilbenes;  4-(N-methylamino)-4'-(2-(2-(2-fluoroethoxy)ethoxy)ethoxy)stilbene;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Comment;  Aniline Compounds;  Deep Learning*;  Magnetic Resonance Imaging*;  Positron-Emission Tomography;  Stilbenes;  Aniline Compounds;  Stilbenes;  4-(N-methylamino)-4'-(2-(2-(2-fluoroethoxy)ethoxy)ethoxy)stilbene;  \"^^xsd:string ;\n",
      "    ns2:Title 30526357 ;\n",
      "    ns4:volume \"290\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://pubs.rsna.org/doi/10.1148/radiol.2018182573?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub  0pubmed']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30526350,26714963,28321471,28434374,22864642,21971456,32451783,32052114,31784863,31028166\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30528509> a 30528509 ;\n",
      "    ns9:abstract \" Cancer samples clustering based on biomolecular data has been becoming an important tool for cancer classification. The recognition of cancer types is of great importance for cancer treatment. In this paper, in order to improve the accuracy of cancer recognition, we propose to use Laplacian regularized Low-Rank Representation (LLRR) to cluster the cancer samples based on genomic data. In LLRR method, the high-dimensional genomic data are approximately treated as samples extracted from a combination of several low-rank subspaces. The purpose of LLRR method is to seek the lowest-rank representation matrix based on a dictionary. Because a Laplacian regularization based on manifold is introduced into LLRR, compared to the Low-Rank Representation (LRR) method, besides capturing the global geometric structure, LLRR can capture the intrinsic local structure of high-dimensional observation data well. And what is more, in LLRR, the original data themselves are selected as a dictionary, so the lowest-rank representation is actually a similar expression between the samples. Therefore, corresponding to the low-rank representation matrix, the samples with high similarity are considered to come from the same subspace and are grouped into a class. The experiment results on real genomic data illustrate that LLRR method, compared with LRR and MLLRR, is more robust to noise and has a better ability to learn the inherent subspace structure of data, and achieves remarkable performance in the clustering of cancer samples. \"^^xsd:string ;\n",
      "    ns8:name \"Juan Wang; Jin-Xing Liu; Xiang-Zhen Kong; Sha-Sha Yuan; Ling-Yun Dai; \"^^xsd:string ;\n",
      "    ns5:journal \"Comput Biol Chem\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Affinity matrix; Gene expression data; Graph regularization; Low-rank representation; Subspace clustering; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Affinity matrix; Gene expression data; Graph regularization; Low-rank representation; Subspace clustering;   Cluster Analysis;  Humans;  Machine Learning*;  Neoplasms / genetics*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Cluster Analysis;  Humans;  Machine Learning*;  Neoplasms / genetics*;  \"^^xsd:string ;\n",
      "    ns2:Title 30528509 ;\n",
      "    ns4:volume \"78\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S1476-9271(18)30785-0']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29990217,27693191,31888442,25073170,25005156,32038712\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30575527> a 30575527 ;\n",
      "    ns9:abstract \" Cough is a protective reflex conveying information on the state of the respiratory system. Cough assessment has been limited so far to subjective measurement tools or uncomfortable (i.e., non-wearable) cough monitors. This limits the potential of real-time cough monitoring to improve respiratory care.   Objective:  This paper presents a machine hearing system for audio-based robust cough segmentation that can be easily deployed in mobile scenarios.   Methods:  Cough detection is performed in two steps. First, a short-term spectral feature set is separately computed in five predefined frequency bands: [0, 0.5), [0.5, 1), [1, 1.5), [1.5, 2), and [2, 5.5125] kHz. Feature selection and combination are then applied to make the short-term feature set robust enough in different noisy scenarios. Second, high-level data representation is achieved by computing the mean and standard deviation of short-term descriptors in 300 ms long-term frames. Finally, cough detection is carried out using a support vector machine trained with data from different noisy scenarios. The system is evaluated using a patient signal database which emulates three real-life scenarios in terms of noise content.   Results:  The system achieves 92.71% sensitivity, 88.58% specificity, and 90.69% Area Under Receiver Operating Charcteristic (ROC) curve (AUC), outperforming state-of-the-art methods.   Conclusion:  Our research outcome paves the way to create a device for cough monitoring in real-life situations.   Significance:  Our proposal is aligned with a more comfortable and less disruptive patient monitoring, with benefits for patients (allows self-monitoring of cough symptoms), practitioners (e.g., assessment of treatments or better clinical understanding of cough patterns), and national health systems (by reducing hospitalizations). \"^^xsd:string ;\n",
      "    ns8:name \"Jesus Monge-Alvarez; Carlos Hoyos-Barcelo; Luis Miguel San-Jose-Revuelta; Pablo Casaseca-de-la-Higuera; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Biomed Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Aged;  Cough* / classification;  Cough* / diagnosis;  Female;  Humans;  Male;  Middle Aged;  Monitoring, Physiologic / methods;  Sensitivity and Specificity;  Signal Processing, Computer-Assisted*;  Signal-To-Noise Ratio;  Sound Spectrography / methods;  Support Vector Machine*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Aged;  Cough* / classification;  Cough* / diagnosis;  Female;  Humans;  Male;  Middle Aged;  Monitoring, Physiologic / methods;  Sensitivity and Specificity;  Signal Processing, Computer-Assisted*;  Signal-To-Noise Ratio;  Sound Spectrography / methods;  Support Vector Machine*;  \"^^xsd:string ;\n",
      "    ns2:Title 30575527 ;\n",
      "    ns4:volume \"66\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TBME.2018.2888998']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29994432,28269103,29993458,24940659,27606394\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30613101> a 30613101 ;\n",
      "    ns9:abstract \"\"^^xsd:string ;\n",
      "    ns8:name \"Rima Arnaout; \"^^xsd:string ;\n",
      "    ns5:journal \"Nat Med\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  News;  Algorithms;  Cardiology;  Deep Learning;  Health*;  Humans;  Radiology;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"News;  Algorithms;  Cardiology;  Deep Learning;  Health*;  Humans;  Radiology;  \"^^xsd:string ;\n",
      "    ns2:Title 30613101 ;\n",
      "    ns4:volume \"25\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://dx.doi.org/10.1038/s41591-018-0318-x']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29290592,29396120,29398492,30783371,29936244\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30617720> a 30617720 ;\n",
      "    ns9:abstract \"  Rationale and objectives:  The ultrasound B-mode-based morphological and texture analysis and Nakagami parametric imaging have been proposed to characterize breast tumors. Since these three feature categories of ultrasonic tissue characterization supply information on different physical characteristics of breast tumors, by combining the above methods is expected to provide more clues for classifying breast tumors.   Materials and methods:  To verify the validity of the concept, raw data were obtained from 160 clinical cases. Six different types of morphological-feature parameters, four texture features, and the Nakagami parameter of benignancy and malignancy were extracted for evaluation. The Pearson's correlation matrix was used to calculate the correlation between different feature parameters. The fuzzy c-means clustering and stepwise regression techniques were utilized to determine the optimal feature set, respectively. The logistic regression, receiver operating characteristic curve, and support vector machine were used to estimate the diagnostic ability.   Results:  The best performance was obtained by combining morphological-feature parameter (e.g., standard deviation of the shortest distance), texture feature (e.g., variance), and the Nakagami parameter, with an accuracy of 89.4%, a specificity of 86.3%, a sensitivity of 92.5%, and an area under receiver operating characteristic curve of 0.96. There was no significant difference between using fuzzy c-means clustering, logistic regression, and support vector machine based on the optimal feature set for breast tumors classification.   Conclusion:  Therefore, we verified that different physical ultrasonic features are functionally complementary and thus improve the performance in diagnosing breast tumors. Moreover, the optimal feature set had the maximum discriminating performance should be irrelative to the power of classifiers. \"^^xsd:string ;\n",
      "    ns8:name \"Soa-Min Hsu; Wen-Hung Kuo; Fang-Chuan Kuo; Yin-Yin Liao; \"^^xsd:string ;\n",
      "    ns5:journal \"Int J Comput Assist Radiol Surg\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Breast ultrasound; Classification; Morphological features; Nakagami parameter; Texture features; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Breast ultrasound; Classification; Morphological features; Nakagami parameter; Texture features;   Breast / diagnostic imaging*;  Breast Neoplasms / classification*;  Breast Neoplasms / diagnosis;  Female;  Humans;  Image Processing, Computer-Assisted / methods*;  Middle Aged;  ROC Curve;  Sensitivity and Specificity;  Support Vector Machine;  Ultrasonography / methods*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Breast / diagnostic imaging*;  Breast Neoplasms / classification*;  Breast Neoplasms / diagnosis;  Female;  Humans;  Image Processing, Computer-Assisted / methods*;  Middle Aged;  ROC Curve;  Sensitivity and Specificity;  Support Vector Machine;  Ultrasonography / methods*;  \"^^xsd:string ;\n",
      "    ns2:Title 30617720 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/s11548-018-01908-8']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"21626954,20129851,22518956,28327449,12467856,31717382\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30624652> a 30624652 ;\n",
      "    ns9:abstract \" Information about the interactions between chemical compounds and proteins is indispensable for understanding the regulation of biological processes and the development of therapeutic drugs. Manually extracting such information from biomedical literature is very time and resource consuming. In this study, we propose a computational method to automatically extract chemical-protein interactions (CPIs) from a given text. Our method extracts CPI pairs and CPI triplets from sentences, where a CPI pair consists of a chemical compound and a protein name, and a CPI triplet consists of a CPI pair along with an interaction word describing their relationship. We extracted a diverse set of features from sentences that were used to build multiple machine learning models. Our models contain both simple features, which can be directly computed from sentences, and more sophisticated features derived using sentence structure analysis techniques. For example, one set of features was extracted based on the shortest paths between the CPI pairs or among the CPI triplets in the dependency graphs obtained from sentence parsing. We designed a three-stage approach to predict the multiple categories of CPIs. Our method performed the best among systems that use non-deep learning methods and outperformed several deep-learning-based systems in the track 5 of the BioCreative VI challenge. The features we designed in this study are informative and can be applied to other machine learning methods including deep learning. \"^^xsd:string ;\n",
      "    ns8:name \"Pei-Yau Lung; Zhe He; Tingting Zhao; Disa Yu; Jinfeng Zhang; \"^^xsd:string ;\n",
      "    ns5:journal \"Database (Oxford)\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, N.I.H., Extramural;  Computational Biology / methods*;  Data Mining / methods*;  Databases, Chemical*;  Databases, Protein*;  Humans;  Machine Learning*;  Pharmaceutical Preparations / chemistry;  Pharmaceutical Preparations / metabolism;  Proteins / chemistry;  Proteins / metabolism;  Semantics;  Software;  Pharmaceutical Preparations;  Proteins;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Computational Biology / methods*;  Data Mining / methods*;  Databases, Chemical*;  Databases, Protein*;  Humans;  Machine Learning*;  Pharmaceutical Preparations / chemistry;  Pharmaceutical Preparations / metabolism;  Proteins / chemistry;  Proteins / metabolism;  Semantics;  Software;  Pharmaceutical Preparations;  Proteins;  \"^^xsd:string ;\n",
      "    ns2:Title 30624652 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://academic.oup.com/database/article-lookup/doi/10.1093/database/bay138', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30624652/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28113437,30689846,27087307,25448299,21369884,31622463\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30632459> a 30632459 ;\n",
      "    ns9:abstract \" Membrane bioreactor (MBR) has been widely employed for industrial effluent treatment, as its higher efficiency in removing pollutants makes effluent reuse more feasible. However, membrane fouling remains as a limiting factor for its greater diffusion. This work performed a sensitivity analysis study to investigate the effects of analytical and operating variables on membrane permeability. The case study is a MBR treating oil refinery effluents. After the identification and validation of a predictive neural model for permeability, sensitivity analysis methods based on both connection weights and variable disturbances were used to quantify and rank the variables influence. A comprehensive analysis showed that Suspended solids and Days between cleanings exerted greater effects on permeability, whereas sludge filterability and sludge temperature were less significant. In sequence, a specific analysis revealed distinct dynamics in MBR operation given different solids concentrations. For instance, from higher solids concentrations, among all the evaluated parameters, only COD presented low significance to the permeability. This evidence suggests that permeability is more susceptible to variations when operating with higher concentrations of Suspended solids. The global result of this study contributes to more efficient MBR operations since distinct relations with permeability imply different effects on membrane fouling. \"^^xsd:string ;\n",
      "    ns8:name \"Aline R Alkmim; Gustavo M de Almeida; Deborah M de Carvalho; Míriam C S Amaral; Sílvia M A C Oliveira; \"^^xsd:string ;\n",
      "    ns5:journal \"Environ Technol\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Membrane bioreactor; artificial neural network; membrane fouling; permeability; sensitivity analysis; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Membrane bioreactor; artificial neural network; membrane fouling; permeability; sensitivity analysis;   Bioreactors*;  Membranes, Artificial*;  Neural Networks, Computer;  Permeability;  Sewage;  Waste Disposal, Fluid;  Membranes, Artificial;  Sewage;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Bioreactors*;  Membranes, Artificial*;  Neural Networks, Computer;  Permeability;  Sewage;  Waste Disposal, Fluid;  Membranes, Artificial;  Sewage;  \"^^xsd:string ;\n",
      "    ns2:Title 30632459 ;\n",
      "    ns4:volume \"41\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://www.tandfonline.com/doi/full/10.1080/09593330.2019.1567609']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"26508453,30014978,32385206,27362445,26776150\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30653830> a 30653830 ;\n",
      "    ns9:abstract \" Specialty crop herbicides are not a priority for the agrochemical industry, and many of these crops do not have access to effective herbicides. High-value fruit and vegetable crops represent small markets and high potential liability in the case of herbicide-induced crop damage. Meanwhile, conventional and organic specialty crop producers are experiencing labor shortages and higher manual weeding costs. Robotic weeders are promising new weed control tools for specialty crops, because they are cheaper to develop and, with fewer environmental and human health risks, are less regulated than herbicides. Now is the time for greater investment in robotic weeders as new herbicides are expensive to develop and few in number, organic crops need better weed control technology and governments are demanding reduced use of pesticides. Public funding of fundamental research on robotic weeder technology can help improve weed and crop recognition, weed control actuators, and expansion of weed science curricula to train students in this technology. Robotic weeders can expand the array of tools available to specialty crop growers. However, the development of robotic weeders will require a broader recognition that these tools are a viable path to create new weed control tools for specialty crops. © 2019 Society of Chemical Industry. \"^^xsd:string ;\n",
      "    ns8:name \"Steven A Fennimore; Matthew Cutulle; \"^^xsd:string ;\n",
      "    ns5:journal \"Pest Manag Sci\"^^xsd:string ;\n",
      "    ns7:key_abs \"  disruptive technology; physical weed control; robotic weeders; robotics; specialty crops; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  disruptive technology; physical weed control; robotic weeders; robotics; specialty crops;   Crop Production / methods*;  Crops, Agricultural;  Herbicides / administration & dosage;  Robotics / instrumentation*;  Weed Control / instrumentation*;  Weed Control / methods*;  Herbicides;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Crop Production / methods*;  Crops, Agricultural;  Herbicides / administration & dosage;  Robotics / instrumentation*;  Weed Control / instrumentation*;  Weed Control / methods*;  Herbicides;  \"^^xsd:string ;\n",
      "    ns2:Title 30653830 ;\n",
      "    ns4:volume \"75\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1002/ps.5337']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"25052888,20586458,23794176,28160383,30652281,31947520\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30670255> a 30670255 ;\n",
      "    ns9:abstract \" Traditional Chinese Medicine (TCM) is an experiential form of medicine with a history dating back thousands of years. The present study aimed to utilize neural network analysis to examine specific prescriptions for colorectal cancer (CRC) in clinical practice to arrive at the most effective prescription strategy. The study analyzed the data of 261 CRC cases recruited from a total of 141,962 cases of renowned veteran TCM doctors collected from datasets of both the DeepMedic software and TCM cancer treatment books. The DeepMedic software was applied to normalize the symptoms/signs and Chinese herbal medicine (CHM) prescriptions using standardized terminologies. Over 20 percent of CRC patients demonstrated symptoms of poor appetite, fatigue, loose stool, and abdominal pain. By analyzing the prescription patterns of CHM, we found that Atractylodes macrocephala (Bai-zhu) and Poria (Fu-ling) were the most commonly prescribed single herbs identified through analysis of medical records, and supported by the neural network analysis; although there was a slight difference in the sequential order. The study revealed an 81.9% degree of similarity of CHM prescriptions between the medical records and the neural network suggestions. The patterns of nourishing Qi and eliminating dampness were the most common goals of clinical prescriptions, which corresponds with treatments of CRC patients in clinical practice. This is the first study to employ machine learning, specifically neural network analytics to support TCM clinical diagnoses and prescriptions. The DeepMedic software may be used to deliver accurate TCM diagnoses and suggest prescriptions to treat CRC. \"^^xsd:string ;\n",
      "    ns8:name \"Yu-Chuan Lin; Wei-Te Huang; Shi-Chen Ou; Hao-Hsiu Hung; Wie-Zen Cheng; Sheng-Shing Lin; Hung-Jen Lin; Sheng-Teng Huang; \"^^xsd:string ;\n",
      "    ns5:journal \"Complement Ther Med\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Chinese herbal medicine; Colorectal cancer; Machine learning; Neural network analysis; Traditional Chinese medicine; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Chinese herbal medicine; Colorectal cancer; Machine learning; Neural network analysis; Traditional Chinese medicine;   Colorectal Neoplasms / drug therapy*;  Databases, Factual;  Drug Prescriptions;  Drugs, Chinese Herbal / therapeutic use*;  Humans;  Medicine, Chinese Traditional / methods;  Neural Networks, Computer;  Software;  Drugs, Chinese Herbal;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Colorectal Neoplasms / drug therapy*;  Databases, Factual;  Drug Prescriptions;  Drugs, Chinese Herbal / therapeutic use*;  Humans;  Medicine, Chinese Traditional / methods;  Neural Networks, Computer;  Software;  Drugs, Chinese Herbal;  \"^^xsd:string ;\n",
      "    ns2:Title 30670255 ;\n",
      "    ns4:volume \"42\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0965-2299(18)31084-7']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"24945402,23947955,25865681,29102765,25115380,32457636\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30678171> a 30678171 ;\n",
      "    ns9:abstract \" As an abundant post-transcriptional modification, dihydrouridine (D) has been found in transfer RNA (tRNA) from bacteria, eukaryotes, and archaea. Nonetheless, knowledge of the exact biochemical roles of dihydrouridine in mediating tRNA function is still limited. Accurate identification of the position of D sites is essential for understanding their functions. Therefore, it is desirable to develop novel methods to identify D sites. In this study, an ensemble classifier was proposed for the detection of D modification sites in the <i>Saccharomyces cerevisiae</i> transcriptome by using heterogeneous features. The jackknife test results demonstrate that the proposed predictor is promising for the identification of D modification sites. It is anticipated that the proposed method can be widely used for identifying D modification sites in tRNA. \"^^xsd:string ;\n",
      "    ns8:name \"Pengmian Feng; Zhaochun Xu; Hui Yang; Hao Lv; Hui Ding; Li Liu; \"^^xsd:string ;\n",
      "    ns5:journal \"Molecules\"^^xsd:string ;\n",
      "    ns7:key_abs \"  RNA secondary structure; dihydrouridine; ensemble classifier; nucleotide physicochemical property; pseudo dinucleotide composition; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  RNA secondary structure; dihydrouridine; ensemble classifier; nucleotide physicochemical property; pseudo dinucleotide composition;   Algorithms;  Chemical Phenomena;  Nucleic Acid Conformation;  RNA, Transfer / chemistry*;  Reproducibility of Results;  Saccharomyces cerevisiae / chemistry*;  Support Vector Machine*;  Uridine / analogs & derivatives;  Uridine / chemistry*;  RNA, Transfer;  Uridine;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Chemical Phenomena;  Nucleic Acid Conformation;  RNA, Transfer / chemistry*;  Reproducibility of Results;  Saccharomyces cerevisiae / chemistry*;  Support Vector Machine*;  Uridine / analogs & derivatives;  Uridine / chemistry*;  RNA, Transfer;  Uridine;  \"^^xsd:string ;\n",
      "    ns2:Title 30678171 ;\n",
      "    ns4:volume \"24\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.mdpi.com/resolver?pii=molecules24030380', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30678171/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"12003496,14970222,25815904,19944101,25625329,31247973\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30682710> a 30682710 ;\n",
      "    ns9:abstract \" In recent years, deep learning has revolutionized the field of machine learning, for computer vision in particular. In this approach, a deep (multilayer) artificial neural network (ANN) is trained, most often in a supervised manner using backpropagation. Vast amounts of labeled training examples are required, but the resulting classification accuracy is truly impressive, sometimes outperforming humans. Neurons in an ANN are characterized by a single, static, continuous-valued activation. Yet biological neurons use discrete spikes to compute and transmit information, and the spike times, in addition to the spike rates, matter. Spiking neural networks (SNNs) are thus more biologically realistic than ANNs, and are arguably the only viable option if one wants to understand how the brain computes at the neuronal description level. The spikes of biological neurons are sparse in time and space, and event-driven. Combined with bio-plausible local learning rules, this makes it easier to build low-power, neuromorphic hardware for SNNs. However, training deep SNNs remains a challenge. Spiking neurons' transfer function is usually non-differentiable, which prevents using backpropagation. Here we review recent supervised and unsupervised methods to train deep SNNs, and compare them in terms of accuracy and computational cost. The emerging picture is that SNNs still lag behind ANNs in terms of accuracy, but the gap is decreasing, and can even vanish on some tasks, while SNNs typically require many fewer operations and are the better candidates to process spatio-temporal data. \"^^xsd:string ;\n",
      "    ns8:name \"Amirhossein Tavanaei; Masoud Ghodrati; Saeed Reza Kheradpisheh; Timothée Masquelier; Anthony Maida; \"^^xsd:string ;\n",
      "    ns5:journal \"Neural Netw\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Biological plausibility; Deep learning; Machine learning; Power-efficient architecture; Spiking neural network; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Biological plausibility; Deep learning; Machine learning; Power-efficient architecture; Spiking neural network;   Review;  Action Potentials* / physiology;  Algorithms;  Brain / physiology;  Deep Learning* / trends;  Humans;  Machine Learning / trends;  Models, Neurological*;  Neural Networks, Computer*;  Neurons / physiology;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Review;  Action Potentials* / physiology;  Algorithms;  Brain / physiology;  Deep Learning* / trends;  Humans;  Machine Learning / trends;  Models, Neurological*;  Neural Networks, Computer*;  Neurons / physiology;  \"^^xsd:string ;\n",
      "    ns2:Title 30682710 ;\n",
      "    ns4:volume \"111\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0893-6080(18)30333-2']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31586857,31593843,19731402,30410432,31726331,32821348,32737332,32708343,32477089,32415108\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30702339> a 30702339 ;\n",
      "    ns9:abstract \"  Purpose:  The influence of preceding load and future perceived wellness of professional soccer players is unexamined. This paper simultaneously evaluates the external load (EL) and internal load (IL) for different time frames in combination with presession wellness to predict future perceived wellness using machine learning techniques.   Methods:  Training and match data were collected from a professional soccer team. The EL was measured using global positioning system technology and accelerometry. The IL was obtained using the rating of perceived exertion multiplied by duration. Predictive models were constructed using gradient-boosted regression trees (GBRT) and one naive baseline method. The individual predictions of future wellness items (ie, fatigue, sleep quality, general muscle soreness, stress levels, and mood) were based on a set of EL and IL indicators in combination with presession wellness. The EL and IL were computed for acute and cumulative time frames. The GBRT model's performance on predicting the reported future wellness was compared with the naive baseline's performance by means of absolute prediction error and effect size.   Results:  The GBRT model outperformed the baseline for the wellness items such as fatigue, general muscle soreness, stress levels, and mood. In addition, only the combination of EL, IL, and presession perceived wellness resulted in nontrivial effects for predicting future wellness. Including the cumulative load did not improve the predictive performances.   Conclusions:  The findings may indicate the importance of including both acute load and presession perceived wellness in a broad monitoring approach in professional soccer. \"^^xsd:string ;\n",
      "    ns8:name \"Tim Op De Beéck; Arne Jaspers; Michel S Brink; Wouter G P Frencken; Filip Staes; Jesse J Davis; Werner F Helsen; \"^^xsd:string ;\n",
      "    ns5:journal \"Int J Sports Physiol Perform\"^^xsd:string ;\n",
      "    ns7:key_abs \"  athlete monitoring; football; global positioning system; predictive modeling; rating of perceived exertion; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  athlete monitoring; football; global positioning system; predictive modeling; rating of perceived exertion;   Accelerometry;  Adult;  Affect;  Athletes*;  Fatigue;  Forecasting*;  Geographic Information Systems;  Health Status*;  Humans;  Machine Learning*;  Male;  Models, Theoretical;  Myalgia;  Physical Conditioning, Human;  Sleep;  Soccer;  Stress, Psychological;  Young Adult;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Accelerometry;  Adult;  Affect;  Athletes*;  Fatigue;  Forecasting*;  Geographic Information Systems;  Health Status*;  Humans;  Machine Learning*;  Male;  Models, Theoretical;  Myalgia;  Physical Conditioning, Human;  Sleep;  Soccer;  Stress, Psychological;  Young Adult;  \"^^xsd:string ;\n",
      "    ns2:Title 30702339 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://journals.humankinetics.com/doi/10.1123/ijspp.2017-0864']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29283691,30569785,27712094,29098658,27459866,32218752,32178370\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30712601> a 30712601 ;\n",
      "    ns9:abstract \"  Background and objective:  Fatty liver disease (FLD) is a common clinical complication; it is associated with high morbidity and mortality. However, an early prediction of FLD patients provides an opportunity to make an appropriate strategy for prevention, early diagnosis and treatment. We aimed to develop a machine learning model to predict FLD that could assist physicians in classifying high-risk patients and make a novel diagnosis, prevent and manage FLD.   Methods:  We included all patients who had an initial fatty liver screening at the New Taipei City Hospital between 1st and 31st December 2009. Classification models such as random forest (RF), Naïve Bayes (NB), artificial neural networks (ANN), and logistic regression (LR) were developed to predict FLD. The area under the receiver operating characteristic curve (ROC) was used to evaluate performances among the four models.   Results:  A total of 577 patients were included in this study; of those 377 patients had fatty liver. The area under the receiver operating characteristic (AUROC) of RF, NB, ANN, and LR with 10 fold-cross validation was 0.925, 0.888, 0.895, and 0.854 respectively. Additionally, The accuracy of RF, NB, ANN, and LR 87.48, 82.65, 81.85, and 76.96%.   Conclusion:  In this study, we developed and compared the four classification models to predict fatty liver disease accurately. However, the random forest model showed higher performance than other classification models. Implementation of a random forest model in the clinical setting could help physicians to stratify fatty liver patients for primary prevention, surveillance, early treatment, and management. \"^^xsd:string ;\n",
      "    ns8:name \"Chieh-Chen Wu; Wen-Chun Yeh; Wen-Ding Hsu; Md Mohaimenul Islam; Phung Anh Alex Nguyen; Tahmina Nasrin Poly; Yao-Chin Wang; Hsuan-Chia Yang; Yu-Chuan Jack Li; \"^^xsd:string ;\n",
      "    ns5:journal \"Comput Methods Programs Biomed\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Classification model; Fatty liver disease; Machine learning; Random forest; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Classification model; Fatty liver disease; Machine learning; Random forest;   Adult;  Aged;  Algorithms*;  Early Diagnosis;  Electronic Health Records;  Fatty Liver* / diagnosis;  Fatty Liver* / prevention & control;  Female;  Forecasting;  Humans;  Machine Learning*;  Male;  Middle Aged;  Neural Networks, Computer;  ROC Curve;  Taiwan;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adult;  Aged;  Algorithms*;  Early Diagnosis;  Electronic Health Records;  Fatty Liver* / diagnosis;  Fatty Liver* / prevention & control;  Female;  Forecasting;  Humans;  Machine Learning*;  Male;  Middle Aged;  Neural Networks, Computer;  ROC Curve;  Taiwan;  \"^^xsd:string ;\n",
      "    ns2:Title 30712601 ;\n",
      "    ns4:volume \"170\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0169-2607(18)31572-4']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29677944,30412613,30914181,30049182,29637384,32690077,32665978,32357880\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30716040> a 30716040 ;\n",
      "    ns9:abstract \" Automatic sleep staging has been often treated as a simple classification problem that aims at determining the label of individual target polysomnography epochs one at a time. In this paper, we tackle the task as a sequence-to-sequence classification problem that receives a sequence of multiple epochs as input and classifies all of their labels at once. For this purpose, we propose a hierarchical recurrent neural network named SeqSleepNet (source code is available at http://github.com/pquochuy/SeqSleepNet). At the epoch processing level, the network consists of a filterbank layer tailored to learn frequency-domain filters for preprocessing and an attention-based recurrent layer designed for short-term sequential modeling. At the sequence processing level, a recurrent layer placed on top of the learned epoch-wise features for long-term modeling of sequential epochs. The classification is then carried out on the output vectors at every time step of the top recurrent layer to produce the sequence of output labels. Despite being hierarchical, we present a strategy to train the network in an end-to-end fashion. We show that the proposed network outperforms the state-of-the-art approaches, achieving an overall accuracy, macro F1-score, and Cohen's kappa of 87.1%, 83.3%, and 0.815 on a publicly available dataset with 200 subjects. \"^^xsd:string ;\n",
      "    ns8:name \"Huy Phan; Fernando Andreotti; Navin Cooray; Oliver Y Chen; Maarten De Vos; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Neural Syst Rehabil Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Algorithms;  Attention;  Databases, Factual;  Electroencephalography / statistics & numerical data;  Electromyography;  Electrooculography;  Humans;  Machine Learning;  Neural Networks, Computer*;  Polysomnography / statistics & numerical data*;  Reproducibility of Results;  Sleep Stages / physiology*;  Software;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Algorithms;  Attention;  Databases, Factual;  Electroencephalography / statistics & numerical data;  Electromyography;  Electrooculography;  Humans;  Machine Learning;  Neural Networks, Computer*;  Polysomnography / statistics & numerical data*;  Reproducibility of Results;  Sleep Stages / physiology*;  Software;  \"^^xsd:string ;\n",
      "    ns2:Title 30716040 ;\n",
      "    ns4:volume \"27\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TNSRE.2019.2896659', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30716040/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31289828,30346277,30440666,31946253,26951412\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30753964> a 30753964 ;\n",
      "    ns9:abstract \" This paper investigates the synchronization issue for a family of time-delayed fractional-order complex dynamical networks (FCDNs) with time delay, unknown bounded uncertainty and disturbance. A novel fractional uncertainty and disturbance estimator (FUDE) based feedback control strategy is proposed to not only synchronize the considered FCDNs but also guaranteeing the precise rejection of unmodelled system uncertainty and external disturbance. Especially, in FUDE-based approach, model uncertainties and external disturbance are integrated as a lumped disturbance and it does not require a completely known system model or a disturbance model. On the other hand, the design algorithm for the proposed control strategy is based on the state-space framework, rather than frequency-based design methodologies in the literature, which helps for predominant comprehension of the inner system behaviour. Also, by the temperance of Lyapunov stability theory and fractional calculus, a set of adequate conditions in the linear matrix inequality framework is obtained, which guarantees the robust synchronization of the closed-loop system. Furthermore, an iterative optimization algorithm is proposed to improve control robustness against the external disturbance and model uncertainties. Finally, two numerical illustrations including financial network model, where the influence of adjustment of macro-economic policies in the entire financial system are given to exhibit the rightness and important features of the acquired theoretical results. \"^^xsd:string ;\n",
      "    ns8:name \"P Selvaraj; O M Kwon; R Sakthivel; \"^^xsd:string ;\n",
      "    ns5:journal \"Neural Netw\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Complex dynamical networks; Disturbance and uncertainty estimator; Fractional-order system; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Complex dynamical networks; Disturbance and uncertainty estimator; Fractional-order system;   Algorithms;  Feedback;  Models, Theoretical*;  Neural Networks, Computer*;  Uncertainty;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Feedback;  Models, Theoretical*;  Neural Networks, Computer*;  Uncertainty;  \"^^xsd:string ;\n",
      "    ns2:Title 30753964 ;\n",
      "    ns4:volume \"112\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0893-6080(19)30020-6']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30827711,29677558,24808207,17565507,27108564\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30794638> a 30794638 ;\n",
      "    ns9:abstract \"  Background:  Rapid antibiotic administration is known to improve sepsis outcomes, however early diagnosis remains challenging due to complex presentation. Our objective was to develop a model using readily available electronic health record (EHR) data capable of recognizing infant sepsis at least 4 hours prior to clinical recognition.   Methods and findings:  We performed a retrospective case control study of infants hospitalized ≥48 hours in the Neonatal Intensive Care Unit (NICU) at the Children's Hospital of Philadelphia between September 2014 and November 2017 who received at least one sepsis evaluation before 12 months of age. We considered two evaluation outcomes as cases: culture positive-positive blood culture for a known pathogen (110 evaluations); and clinically positive-negative cultures but antibiotics administered for ≥120 hours (265 evaluations). Case data was taken from the 44-hour window ending 4 hours prior to evaluation. We randomly sampled 1,100 44-hour windows of control data from all times ≥10 days removed from any evaluation. Model inputs consisted of up to 36 features derived from routine EHR data. Using 10-fold nested cross-validation, 8 machine learning models were trained to classify inputs as sepsis positive or negative. When tasked with discriminating culture positive cases from controls, 6 models achieved a mean area under the receiver operating characteristic (AUC) between 0.80-0.82 with no significant differences between them. Including both culture and clinically positive cases, the same 6 models achieved an AUC between 0.85-0.87, again with no significant differences.   Conclusions:  Machine learning models can identify infants with sepsis in the NICU hours prior to clinical recognition. Learning curves indicate model improvement may be achieved with additional training examples. Additional input features may also improve performance. Further research is warranted to assess potential performance improvements and clinical efficacy in a prospective trial. \"^^xsd:string ;\n",
      "    ns8:name \"Aaron J Masino; Mary Catherine Harris; Daniel Forsyth; Svetlana Ostapenko; Lakshmi Srinivasan; Christopher P Bonafide; Fran Balamuth; Melissa Schmatz; Robert W Grundmeier; \"^^xsd:string ;\n",
      "    ns5:journal \"PLoS One\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Critical Care*;  Diagnosis, Computer-Assisted*;  Electronic Health Records*;  Female;  Humans;  Infant;  Infant, Newborn;  Machine Learning*;  Male;  Models, Biological*;  Retrospective Studies;  Sepsis / diagnosis*;  Sepsis / therapy;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Critical Care*;  Diagnosis, Computer-Assisted*;  Electronic Health Records*;  Female;  Humans;  Infant;  Infant, Newborn;  Machine Learning*;  Male;  Models, Biological*;  Retrospective Studies;  Sepsis / diagnosis*;  Sepsis / therapy;  \"^^xsd:string ;\n",
      "    ns2:Title 30794638 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.plos.org/10.1371/journal.pone.0212665', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30794638/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"24043317,30646095,31035074,26251506,23353941,32734166,32518246,32013925,31733815\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30817960> a 30817960 ;\n",
      "    ns9:abstract \"\"^^xsd:string ;\n",
      "    ns8:name \"Riccardo Bertolo; Alessandro Antonelli; Marco Sandri; Pierluigi Bove; Luca Cindolo; Andrea Minervini; AGILE Group (Italian Group For Advanced Laparo-Endoscopic Surgery); \"^^xsd:string ;\n",
      "    ns5:journal \"Urology\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Letter;  Comment;  Laparoscopy*;  Nephrectomy;  Prospective Studies;  Robotics*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Letter;  Comment;  Laparoscopy*;  Nephrectomy;  Prospective Studies;  Robotics*;  \"^^xsd:string ;\n",
      "    ns2:Title 30817960 ;\n",
      "    ns4:volume \"128\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0090-4295(19)30208-0']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30659901,31364939,30659901,22957660,24231845,21610276\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30834812> a 30834812 ;\n",
      "    ns9:abstract \" This study examined the treatment effects between unilateral hybrid therapy (UHT; unilateral robot-assisted therapy [RT] + modified constraint-induced movement therapy) and bilateral hybrid therapy (BHT; bilateral RT + bilateral arm training) compared with RT. Thirty patients with chronic stroke were randomized to UHT, BHT, or RT groups. Preliminary efficacy was assessed using the Fugl-Meyer Assessment (FMA), the Chedoke Arm and Hand Activity Inventory (CAHAI), and the goal attainment scaling (GAS). Possible adverse effects of abnormal muscle tone, pain, and fatigue were recorded. All groups showed large improvements in motor recovery and individual goals. Significant between-group differences were found on GAS favoring the hybrid groups but not on FMA and CAHAI. No adverse effects were reported. Hybrid therapies are safe and applicable interventions for chronic stroke and favorable for improving individual functional goals. Treatment effects on motor recovery and functional activity might be similar among the three groups. \"^^xsd:string ;\n",
      "    ns8:name \"Chung-Shan Hung; Yu-Wei Hsieh; Ching-Yi Wu; Yu-Ju Chen; Keh-Chung Lin; Chia-Ling Chen; Kaiping Grace Yao; Chien-Ting Liu; Yi-Shiung Horng; \"^^xsd:string ;\n",
      "    ns5:journal \"OTJR (Thorofare N J)\"^^xsd:string ;\n",
      "    ns7:key_abs \"  occupational therapy; randomized controlled trial; rehabilitation; stroke; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  occupational therapy; randomized controlled trial; rehabilitation; stroke;   Randomized Controlled Trial;  Research Support, Non-U.S. Gov't;  Female;  Humans;  Male;  Middle Aged;  Occupational Therapy;  Paresis / rehabilitation*;  Physical Therapy Modalities;  Recovery of Function;  Robotics*;  Single-Blind Method;  Stroke Rehabilitation*;  Task Performance and Analysis;  Treatment Outcome;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Randomized Controlled Trial;  Research Support, Non-U.S. Gov't;  Female;  Humans;  Male;  Middle Aged;  Occupational Therapy;  Paresis / rehabilitation*;  Physical Therapy Modalities;  Recovery of Function;  Robotics*;  Single-Blind Method;  Stroke Rehabilitation*;  Task Performance and Analysis;  Treatment Outcome;  \"^^xsd:string ;\n",
      "    ns2:Title 30834812 ;\n",
      "    ns4:volume \"39\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://journals.sagepub.com/doi/10.1177/1539449218825438?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub  0pubmed']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31421096,17270510,30686326,27597165,28540536,32429562\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30840663> a 30840663 ;\n",
      "    ns9:abstract \" Recognizing a person in a crowded environment is a challenging, yet critical, visual-search task for both humans and machine-vision algorithms. This paper explores the possibility of combining a residual neural network (ResNet), brain-computer interfaces (BCIs) and human participants to create \\\"cyborgs\\\" that improve decision making. Human participants and a ResNet undertook the same face-recognition experiment. BCIs were used to decode the decision confidence of humans from their EEG signals. Different types of cyborg groups were created, including either only humans (with or without the BCI) or groups of humans and the ResNet. Cyborg groups decisions were obtained weighing individual decisions by confidence estimates. Results show that groups of cyborgs are significantly more accurate (up to 35%) than the ResNet, the average participant, and equally-sized groups of humans not assisted by technology. These results suggest that melding humans, BCI, and machine-vision technology could significantly improve decision-making in realistic scenarios. \"^^xsd:string ;\n",
      "    ns8:name \"Davide Valeriani; Riccardo Poli; \"^^xsd:string ;\n",
      "    ns5:journal \"PLoS One\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Research Support, U.S. Gov't, Non-P.H.S.;  Adult;  Brain / physiology;  Brain-Computer Interfaces*;  Decision Making*;  Electroencephalography;  Evoked Potentials, Visual / physiology;  Facial Recognition / physiology*;  Female;  Healthy Volunteers;  Humans;  Male;  Neural Networks, Computer*;  Reaction Time / physiology;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Research Support, U.S. Gov't, Non-P.H.S.;  Adult;  Brain / physiology;  Brain-Computer Interfaces*;  Decision Making*;  Electroencephalography;  Evoked Potentials, Visual / physiology;  Facial Recognition / physiology*;  Female;  Healthy Volunteers;  Humans;  Male;  Neural Networks, Computer*;  Reaction Time / physiology;  \"^^xsd:string ;\n",
      "    ns2:Title 30840663 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.plos.org/10.1371/journal.pone.0212935', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30840663/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30897153,29932424,30279309,28225794,29488902,31729671,30897153\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30852421> a 30852421 ;\n",
      "    ns9:abstract \"  Background:  Recent studies have demonstrated the use of convolutional neural networks (CNNs) to classify images of melanoma with accuracies comparable to those achieved by board-certified dermatologists. However, the performance of a CNN exclusively trained with dermoscopic images in a clinical image classification task in direct competition with a large number of dermatologists has not been measured to date. This study compares the performance of a convolutional neuronal network trained with dermoscopic images exclusively for identifying melanoma in clinical photographs with the manual grading of the same images by dermatologists.   Methods:  We compared automatic digital melanoma classification with the performance of 145 dermatologists of 12 German university hospitals. We used methods from enhanced deep learning to train a CNN with 12,378 open-source dermoscopic images. We used 100 clinical images to compare the performance of the CNN to that of the dermatologists. Dermatologists were compared with the deep neural network in terms of sensitivity, specificity and receiver operating characteristics.   Findings:  The mean sensitivity and specificity achieved by the dermatologists with clinical images was 89.4% (range: 55.0%-100%) and 64.4% (range: 22.5%-92.5%). At the same sensitivity, the CNN exhibited a mean specificity of 68.2% (range 47.5%-86.25%). Among the dermatologists, the attendings showed the highest mean sensitivity of 92.8% at a mean specificity of 57.7%. With the same high sensitivity of 92.8%, the CNN had a mean specificity of 61.1%.   Interpretation:  For the first time, dermatologist-level image classification was achieved on a clinical image classification task without training on clinical images. The CNN had a smaller variance of results indicating a higher robustness of computer vision compared with human assessment for dermatologic image classification tasks. \"^^xsd:string ;\n",
      "    ns8:name \"Titus J Brinker; Achim Hekler; Alexander H Enk; Joachim Klode; Axel Hauschild; Carola Berking; Bastian Schilling; Sebastian Haferkamp; Dirk Schadendorf; Stefan Fröhling; Jochen S Utikal; Christof von Kalle; Collaborators; Wiebke Ludwig-Peitsch; Judith Sirokay; Lucie Heinzerling; Magarete Albrecht; Katharina Baratella; Lena Bischof; Eleftheria Chorti; Anna Dith; Christina Drusio; Nina Giese; Emmanouil Gratsias; Klaus Griewank; Sandra Hallasch; Zdenka Hanhart; Saskia Herz; Katja Hohaus; Philipp Jansen; Finja Jockenhöfer; Theodora Kanaki; Sarah Knispel; Katja Leonhard; Anna Martaki; Liliana Matei; Johanna Matull; Alexandra Olischewski; Maximilian Petri; Jan-Malte Placke; Simon Raub; Katrin Salva; Swantje Schlott; Elsa Sody; Nadine Steingrube; Ingo Stoffels; Selma Ugurel; Wiebke Sondermann; Anne Zaremba; Christoffer Gebhardt; Nina Booken; Maria Christolouka; Kristina Buder-Bakhaya; Therezia Bokor-Billmann; Alexander Enk; Patrick Gholam; Holger Hänßle; Martin Salzmann; Sarah Schäfer; Knut Schäkel; Timo Schank; Ann-Sophie Bohne; Sophia Deffaa; Katharina Drerup; Friederike Egberts; Anna-Sophie Erkens; Benjamin Ewald; Sandra Falkvoll; Sascha Gerdes; Viola Harde; Marion Jost; Katja Kosova; Laetitia Messinger; Malte Metzner; Kirsten Morrison; Rogina Motamedi; Anja Pinczker; Anne Rosenthal; Natalie Scheller; Thomas Schwarz; Dora Stölzl; Federieke Thielking; Elena Tomaschewski; Ulrike Wehkamp; Michael Weichenthal; Oliver Wiedow; Claudia Maria Bär; Sophia Bender-Säbelkampf; Marc Horbrügger; Ante Karoglan; Luise Kraas; Jörg Faulhaber; Cyrill Geraud; Ze Guo; Philipp Koch; Miriam Linke; Nolwenn Maurier; Verena Müller; Benjamin Thomas; Jochen Sven Utikal; Ali Saeed M Alamri; Andrea Baczako; Matthias Betke; Carolin Haas; Daniela Hartmann; Markus V Heppt; Katharina Kilian; Sebastian Krammer; Natalie Lidia Lapczynski; Sebastian Mastnik; Suzan Nasifoglu; Cristel Ruini; Elke Sattler; Max Schlaak; Hans Wolff; Birgit Achatz; Astrid Bergbreiter; Konstantin Drexler; Monika Ettinger; Anna Halupczok; Marie Hegemann; Verena Dinauer; Maria Maagk; Marion Mickler; Biance Philipp; Anna Wilm; Constanze Wittmann; Anja Gesierich; Valerie Glutsch; Katrin Kahlert; Andreas Kerstan; Philipp Schrüfer; \"^^xsd:string ;\n",
      "    ns5:journal \"Eur J Cancer\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Artificial intelligence; Diagnostics; Melanoma; Skin cancer; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Artificial intelligence; Diagnostics; Melanoma; Skin cancer;   Dermatologists;  Dermatology / methods;  Dermoscopy / methods*;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Melanoma / classification*;  Neural Networks, Computer*;  Sensitivity and Specificity;  Skin Neoplasms / classification*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Dermatologists;  Dermatology / methods;  Dermoscopy / methods*;  Humans;  Image Interpretation, Computer-Assisted / methods*;  Melanoma / classification*;  Neural Networks, Computer*;  Sensitivity and Specificity;  Skin Neoplasms / classification*;  \"^^xsd:string ;\n",
      "    ns2:Title 30852421 ;\n",
      "    ns4:volume \"111\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0959-8049(19)30144-3']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30981091,31401469,31419752,31017580,31255749,32617318,32596246,32471471,32253623,31799995\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30861015> a 30861015 ;\n",
      "    ns9:abstract \" Temperament and Psychological Types can be defined as innate psychological characteristics associated with how we relate with the world, and often influence our study and career choices. Furthermore, understanding these features help us manage conflicts, develop leadership, improve teaching and many other skills. Assigning temperament and psychological types is usually made by filling specific questionnaires. However, it is possible to identify temperamental characteristics from a linguistic and behavioral analysis of social media data from a user. Thus, machine-learning algorithms can be used to learn from a user's social media data and infer his/her behavioral type. This paper initially provides a brief historical review of theories on temperament and then brings a survey of research aimed at predicting temperament and psychological types from social media data. It follows with the proposal of a framework to predict temperament and psychological types from a linguistic and behavioral analysis of Twitter data. The proposed framework infers temperament types following the David Keirsey's model, and psychological types based on the MBTI model. Various data modelling and classifiers are used. The results showed that Random Forests with the LIWC technique can predict with 96.46% of accuracy the Artisan temperament, 92.19% the Guardian temperament, 78.68% the Idealist, and 83.82% the Rational temperament. The MBTI results also showed that Random Forests achieved a better performance with an accuracy of 82.05% for the E/I pair, 88.38% for the S/N pair, 80.57% for the T/F pair, and 78.26% for the J/P pair. \"^^xsd:string ;\n",
      "    ns8:name \"Ana Carolina E S Lima; Leandro Nunes de Castro; \"^^xsd:string ;\n",
      "    ns5:journal \"PLoS One\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Historical Article;  Research Support, Non-U.S. Gov't;  Review;  Behavioral Research / history;  Behavioral Research / methods*;  Female;  History, 21st Century;  Humans;  Machine Learning / history;  Male;  Models, Psychological;  Psycholinguistics / history;  Psycholinguistics / methods*;  Social Behavior*;  Social Media*;  Temperament*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Historical Article;  Research Support, Non-U.S. Gov't;  Review;  Behavioral Research / history;  Behavioral Research / methods*;  Female;  History, 21st Century;  Humans;  Machine Learning / history;  Male;  Models, Psychological;  Psycholinguistics / history;  Psycholinguistics / methods*;  Social Behavior*;  Social Media*;  Temperament*;  \"^^xsd:string ;\n",
      "    ns2:Title 30861015 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.plos.org/10.1371/journal.pone.0212844', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/30861015/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"26307512,24684522,28694239,24929463,21978734\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30876845> a 30876845 ;\n",
      "    ns9:abstract \" Machine learning, especially deep learning, has the predictive power to predict adverse drug reactions, repurpose drugs and perform precision medicine. We provide a background of machine learning and propose a potential high-performance deep learning framework for its successful applications in these practices. \"^^xsd:string ;\n",
      "    ns8:name \"Chun Yen Lee; Yi-Ping Phoebe Chen; \"^^xsd:string ;\n",
      "    ns5:journal \"Drug Discov Today\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Systematic Review;  Drug-Related Side Effects and Adverse Reactions / prevention & control*;  Humans;  Machine Learning*;  Pharmacovigilance*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Systematic Review;  Drug-Related Side Effects and Adverse Reactions / prevention & control*;  Humans;  Machine Learning*;  Pharmacovigilance*;  \"^^xsd:string ;\n",
      "    ns2:Title 30876845 ;\n",
      "    ns4:volume \"24\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S1359-6446(18)30367-2']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31383376,32570592,30762164,31200909,26654809,31868917\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/30941889> a 30941889 ;\n",
      "    ns9:abstract \" Annotation of protein functions plays an important role in understanding life at the molecular level. High-throughput sequencing produces massive numbers of raw proteins sequences and only about 1% of them have been manually annotated with functions. Experimental annotations of functions are expensive, time-consuming and do not keep up with the rapid growth of the sequence numbers. This motivates the development of computational approaches that predict protein functions. A novel deep learning framework, DeepFunc, is proposed which accurately predicts protein functions from protein sequence- and network-derived information. More precisely, DeepFunc uses a long and sparse binary vector to encode information concerning domains, families, and motifs collected from the InterPro tool that is associated with the input protein sequence. This vector is processed with two neural layers to obtain a low-dimensional vector which is combined with topological information extracted from protein-protein interactions (PPIs) and functional linkages. The combined information is processed by a deep neural network that predicts protein functions. DeepFunc is empirically and comparatively tested on a benchmark testing dataset and the Critical Assessment of protein Function Annotation algorithms (CAFA) 3 dataset. The experimental results demonstrate that DeepFunc outperforms current methods on the testing dataset and that it secures the highest F<sub>max</sub> = 0.54 and AUC = 0.94 on the CAFA3 dataset. \"^^xsd:string ;\n",
      "    ns8:name \"Fuhao Zhang; Hong Song; Min Zeng; Yaohang Li; Lukasz Kurgan; Min Li; \"^^xsd:string ;\n",
      "    ns5:journal \"Proteomics\"^^xsd:string ;\n",
      "    ns7:key_abs \"  deep learning; functional linkages; protein domains; protein functions; protein sequences; protein-protein interactions; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  deep learning; functional linkages; protein domains; protein functions; protein sequences; protein-protein interactions;   Research Support, Non-U.S. Gov't;  Algorithms;  Amino Acid Sequence / genetics*;  Computational Biology;  Deep Learning;  High-Throughput Nucleotide Sequencing;  Humans;  Molecular Sequence Annotation*;  Neural Networks, Computer;  Proteins / genetics*;  Proteins;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Algorithms;  Amino Acid Sequence / genetics*;  Computational Biology;  Deep Learning;  High-Throughput Nucleotide Sequencing;  Humans;  Molecular Sequence Annotation*;  Neural Networks, Computer;  Proteins / genetics*;  Proteins;  \"^^xsd:string ;\n",
      "    ns2:Title 30941889 ;\n",
      "    ns4:volume \"19\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1002/pmic.201900019']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31187588,31187588,31221760,31298436,27879651,25548927,32787776,31787076\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31022847> a 31022847 ;\n",
      "    ns9:abstract \" Everyday, people interact with different types of human machine interfaces, and the use of them is increasing, thus, it is necessary to design interfaces which are capable of responding in an intelligent, natural, inexpensive, and accessible way, regardless of social, cultural, economic, or physical features of a user. In this sense, it has been sought out the development of small interfaces to avoid any type of user annoyance. In this paper, bioelectric signals have been analyzed and characterized in order to propose a more natural human-machine interaction system. The proposed scheme is controlled by electromyographic signals that a person can create through arm movements. Such arm signals have been analyzed and characterized by a back-propagation neural network, and by a wavelet analysis, in this way control commands were obtained from such arm electromyographic signals. The developed interface, uses Extensible Messaging and Presence Protocol (XMPP) to send control commands remotely. In the experiment, it manipulated a vehicle that was approximately 52 km away from the user, with which it can be showed that a characterized electromyographic signal can be sufficient for controlling embedded devices such as a Raspberri Pi, and in this way we can use the neural network and the wavelet analysis to generate control words which can be used inside the Internet of Things too. A Tiva-C board has been used to acquire data instead of more popular development boards, with an adequate response. One of the most important aspects related to the proposed interface is that it can be used by almost anyone, including people with different abilities and even illiterate people. Due to the existence of individual efforts to characterize different types of bioelectric signals, we propose the generation of free access Bioelectric Control Dictionary, to define and consult each characterized biosignal. \"^^xsd:string ;\n",
      "    ns8:name \"David Tinoco Varela; Fernando Gudiño Peñaloza; Carolina Jeanette Villaseñor Rodelas; \"^^xsd:string ;\n",
      "    ns5:journal \"Sensors (Basel)\"^^xsd:string ;\n",
      "    ns7:key_abs \"  biosignals; command and control systems; human machine interfaces; integrated circuit interconnections; intelligent control; nano systems; wavelets; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  biosignals; command and control systems; human machine interfaces; integrated circuit interconnections; intelligent control; nano systems; wavelets;   Algorithms;  Humans;  Man-Machine Systems;  Neural Networks, Computer*;  User-Computer Interface*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Humans;  Man-Machine Systems;  Neural Networks, Computer*;  User-Computer Interface*;  \"^^xsd:string ;\n",
      "    ns2:Title 31022847 ;\n",
      "    ns4:volume \"19\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.mdpi.com/resolver?pii=s19081923', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31022847/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"17945910,19163712,30337084,19665554,22438708\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31029054> a 31029054 ;\n",
      "    ns9:abstract \" This paper deals with the fixed-time synchronization problem of coupled delayed neural networks with discontinuous activations. Based on pinning control, a discontinuous controller is firstly proposed to guarantee that coupled neural networks achieve synchronization with a desired trajectory in finite time. Then, a discontinuous fixed-time controller is designed. With the fixed-time controller, the settling time can be estimated regardless of initial conditions. By providing a topology-dependent Lyapunov function, some criteria of finite-/fixed-time synchronization are derived. Finally, two numerical examples are given to show the effectiveness of the proposed controllers. \"^^xsd:string ;\n",
      "    ns8:name \"Hui Lü; Wangli He; Qing-Long Han; Chen Peng; \"^^xsd:string ;\n",
      "    ns5:journal \"Neural Netw\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Discontinuous activations; Fixed-time synchronization; Neural networks; Pinning control; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Discontinuous activations; Fixed-time synchronization; Neural networks; Pinning control;   Algorithms;  Cortical Synchronization / physiology*;  Neural Networks, Computer*;  Time Factors;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Cortical Synchronization / physiology*;  Neural Networks, Computer*;  Time Factors;  \"^^xsd:string ;\n",
      "    ns2:Title 31029054 ;\n",
      "    ns4:volume \"116\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0893-6080(19)30108-X']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28110107,29414532,28388472,23115594,28364661\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31029061> a 31029061 ;\n",
      "    ns9:abstract \"  Background:  The complex nature of biological data has driven the development of specialized software tools. Scientific workflow management systems simplify the assembly of such tools into pipelines, assist with job automation, and aid reproducibility of analyses. Many contemporary workflow tools are specialized or not designed for highly complex workflows, such as with nested loops, dynamic scheduling, and parametrization, which is common in, e.g., machine learning.   Findings:  SciPipe is a workflow programming library implemented in the programming language Go, for managing complex and dynamic pipelines in bioinformatics, cheminformatics, and other fields. SciPipe helps in particular with workflow constructs common in machine learning, such as extensive branching, parameter sweeps, and dynamic scheduling and parametrization of downstream tasks. SciPipe builds on flow-based programming principles to support agile development of workflows based on a library of self-contained, reusable components. It supports running subsets of workflows for improved iterative development and provides a data-centric audit logging feature that saves a full audit trace for every output file of a workflow, which can be converted to other formats such as HTML, TeX, and PDF on demand. The utility of SciPipe is demonstrated with a machine learning pipeline, a genomics, and a transcriptomics pipeline.   Conclusions:  SciPipe provides a solution for agile development of complex and dynamic pipelines, especially in machine learning, through a flexible application programming interface suitable for scientists used to programming or scripting. \"^^xsd:string ;\n",
      "    ns8:name \"Samuel Lampa; Martin Dahlö; Jonathan Alvarsson; Ola Spjuth; \"^^xsd:string ;\n",
      "    ns5:journal \"Gigascience\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Go; Golang; flow-based programming; machine learning; pipelines; reproducibility; scientific workflow management systems; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Go; Golang; flow-based programming; machine learning; pipelines; reproducibility; scientific workflow management systems;   Research Support, Non-U.S. Gov't;  Computational Biology*;  Gene Library;  Genomics*;  Machine Learning;  Programming Languages;  Software*;  Workflow;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Computational Biology*;  Gene Library;  Genomics*;  Machine Learning;  Programming Languages;  Software*;  Workflow;  \"^^xsd:string ;\n",
      "    ns2:Title 31029061 ;\n",
      "    ns4:volume \"8\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://academic.oup.com/gigascience/article-lookup/doi/10.1093/gigascience/giz044', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31029061/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"22594911,22500002,21352538,30716464,31510695,32256520\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31039552> a 31039552 ;\n",
      "    ns9:abstract \"  Objective:  We investigate the biometric specificity of the frequency following response (FFR), an EEG marker of early auditory processing that reflects phase-locked activity from neural ensembles in the auditory cortex and subcortex (Chandrasekaran and Kraus 2010, Bidelman, 2015a, 2018, Coffey et al 2017b). Our objective is two-fold: demonstrate that the FFR contains information beyond stimulus properties and broad group-level markers, and to assess the practical viability of the FFR as a biometric across different sounds, auditory experiences, and recording days.   Approach:  We trained the hidden Markov model (HMM) to decode listener identity from FFR spectro-temporal patterns across multiple frequency bands. Our dataset included FFRs from twenty native speakers of English or Mandarin Chinese (10 per group) listening to Mandarin Chinese tones across three EEG sessions separated by days. We decoded subject identity within the same auditory context (same tone and session) and across different stimuli and recording sessions.   Main results:  The HMM decoded listeners for averaging sizes as small as one single FFR. However, model performance improved for larger averaging sizes (e.g. 25 FFRs), similarity in auditory context (same tone and day), and lack of familiarity with the sounds (i.e. native English relative to native Chinese listeners). Our results also revealed important biometric contributions from frequency bands in the cortical and subcortical EEG.   Significance:  Our study provides the first deep and systematic biometric characterization of the FFR and provides the basis for biometric identification systems incorporating this neural signal. \"^^xsd:string ;\n",
      "    ns8:name \"Fernando Llanos; Zilong Xie; Bharath Chandrasekaran; \"^^xsd:string ;\n",
      "    ns5:journal \"J Neural Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Adult;  Auditory Perception / physiology;  Biometric Identification / methods*;  Electroencephalography / methods*;  Female;  Humans;  Language*;  Machine Learning*;  Male;  Speech / physiology*;  Speech Perception / physiology*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adult;  Auditory Perception / physiology;  Biometric Identification / methods*;  Electroencephalography / methods*;  Female;  Humans;  Language*;  Machine Learning*;  Male;  Speech / physiology*;  Speech Perception / physiology*;  \"^^xsd:string ;\n",
      "    ns2:Title 31039552 ;\n",
      "    ns4:volume \"16\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1088/1741-2552/ab1e01', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31039552/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28807860,29604459,26146197,28638700,11677406\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31094833> a 31094833 ;\n",
      "    ns9:abstract \"  Background:  Identifying patients at risk of not achieving meaningful gains in long-term postsurgical patient-reported outcome measures (PROMs) is important for improving patient monitoring and facilitating presurgical decision support. Machine learning may help automatically select and weigh many predictors to create models that maximize predictive power. However, these techniques are underused among studies of total joint arthroplasty (TJA) patients, particularly those exploring changes in postsurgical PROMs. QUESTION/PURPOSES: (1) To evaluate whether machine learning algorithms, applied to hospital registry data, could predict patients who would not achieve a minimally clinically important difference (MCID) in four PROMs 2 years after TJA; (2) to explore how predictive ability changes as more information is included in modeling; and (3) to identify which variables drive the predictive power of these models.   Methods:  Data from a single, high-volume institution's TJA registry were used for this study. We identified 7239 hip and 6480 knee TJAs between 2007 and 2012, which, for at least one PROM, patients had completed both baseline and 2-year followup surveys (among 19,187 TJAs in our registry and 43,313 total TJAs). In all, 12,203 registry TJAs had valid SF-36 physical component scores (PCS) and mental component scores (MCS) at baseline and 2 years; 7085 and 6205 had valid Hip and Knee Disability and Osteoarthritis Outcome Scores for joint replacement (HOOS JR and KOOS JR scores), respectively. Supervised machine learning refers to a class of algorithms that links a mapping of inputs to an output based on many input-output examples. We trained three of the most popular such algorithms (logistic least absolute shrinkage and selection operator (LASSO), random forest, and linear support vector machine) to predict 2-year postsurgical MCIDs. We incrementally considered predictors available at four time points: (1) before the decision to have surgery, (2) before surgery, (3) before discharge, and (4) immediately after discharge. We evaluated the performance of each model using area under the receiver operating characteristic (AUROC) statistics on a validation sample composed of a random 20% subsample of TJAs excluded from modeling. We also considered abbreviated models that only used baseline PROMs and procedure as predictors (to isolate their predictive power). We further directly evaluated which variables were ranked by each model as most predictive of 2-year MCIDs.   Results:  The three machine learning algorithms performed in the poor-to-good range for predicting 2-year MCIDs, with AUROCs ranging from 0.60 to 0.89. They performed virtually identically for a given PROM and time point. AUROCs for the logistic LASSO models for predicting SF-36 PCS 2-year MCIDs at the four time points were: 0.69, 0.78, 0.78, and 0.78, respectively; for SF-36 MCS 2-year MCIDs, AUROCs were: 0.63, 0.89, 0.89, and 0.88; for HOOS JR 2-year MCIDs: 0.67, 0.78, 0.77, and 0.77; for KOOS JR 2-year MCIDs: 0.61, 0.75, 0.75, and 0.75. Before-surgery models performed in the fair-to-good range and consistently ranked the associated baseline PROM as among the most important predictors. Abbreviated LASSO models performed worse than the full before-surgery models, though they retained much of the predictive power of the full before-surgery models.   Conclusions:  Machine learning has the potential to improve clinical decision-making and patient care by helping to prioritize resources for postsurgical monitoring and informing presurgical discussions of likely outcomes of TJA. Applied to presurgical registry data, such models can predict, with fair-to-good ability, 2-year postsurgical MCIDs. Although we report all parameters of our best-performing models, they cannot simply be applied off-the-shelf without proper testing. Our analyses indicate that machine learning holds much promise for predicting orthopaedic outcomes. LEVEL OF EVIDENCE: Level III, diagnostic study. \"^^xsd:string ;\n",
      "    ns8:name \"Mark Alan Fontana; Stephen Lyman; Gourab K Sarker; Douglas E Padgett; Catherine H MacLean; \"^^xsd:string ;\n",
      "    ns5:journal \"Clin Orthop Relat Res\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Aged;  Arthroplasty, Replacement, Hip*;  Arthroplasty, Replacement, Knee*;  Disability Evaluation;  Female;  Humans;  Machine Learning*;  Male;  Middle Aged;  Minimal Clinically Important Difference*;  Patient Reported Outcome Measures;  Predictive Value of Tests;  Registries;  Retrospective Studies;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Aged;  Arthroplasty, Replacement, Hip*;  Arthroplasty, Replacement, Knee*;  Disability Evaluation;  Female;  Humans;  Machine Learning*;  Male;  Middle Aged;  Minimal Clinically Important Difference*;  Patient Reported Outcome Measures;  Predictive Value of Tests;  Registries;  Retrospective Studies;  \"^^xsd:string ;\n",
      "    ns2:Title 31094833 ;\n",
      "    ns4:volume \"477\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://Insights.ovid.com/pubmed?pmid=31094833', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31094833/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"32195714,30179951,31136425,26956248,29553663,28678462,32332244,32282532,32271173,32257027,32195714\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31122081> a 31122081 ;\n",
      "    ns9:abstract \" Involvement from the dermatology community in the current revolution in sensors, personal health devices, computing, and machine learning algorithms can enable us to reach the promise of precision medicine in dermatology to deliver the 'right treatment at the right time'. Machine learning combined with digital health technology offers enormous potential to expand access to dermatological services, engage patients in the management of their conditions, and provide tools that enable clinicians to be more effective in their delivery of health care. We offer our framework for how machine learning, rather than replacing dermatologists, will be essential in realizing the promise of precision dermatology. \"^^xsd:string ;\n",
      "    ns8:name \"Shannon Wongvibulsin; Byron Kalm-Tsun Ho; Shawn G Kwatra; \"^^xsd:string ;\n",
      "    ns5:journal \"J Dermatolog Treat\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Machine learning; digital health; increasing access; patient engagement; precision dermatology; precision medicine; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Machine learning; digital health; increasing access; patient engagement; precision dermatology; precision medicine;   Algorithms;  Clinical Decision-Making;  Delivery of Health Care*;  Humans;  Machine Learning*;  Precision Medicine;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Clinical Decision-Making;  Delivery of Health Care*;  Humans;  Machine Learning*;  Precision Medicine;  \"^^xsd:string ;\n",
      "    ns2:Title 31122081 ;\n",
      "    ns4:volume \"31\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://www.tandfonline.com/doi/full/10.1080/09546634.2019.1623373']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30953518,32253623,28268284,30102808,28545640\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31135114> a 31135114 ;\n",
      "    ns9:abstract \" Objective To establish a regression algorithm model that applies to bone age estimation of Xinjiang Uygur adolescents with machine learning methods such as histogram of oriented gradient （HOG）, local binary patterns （LBP）, support vector machine （SVM）, principal component analysis （PCA）. Methods DR images of knee-joints from 275 male and 225 female subjects aged 12.0-&lt;19.0 years old were collected, PCA method was used to reduce the dimensionality of the HOG and LBP features, and support vector regression （SVR） was used to establish a knee-joint bone age estimation algorithm model. Stratified random sampling method was used to select 215 male samples, 180 female samples for the model training set. <i>K</i>-fold cross validation method was used to optimize parameters of the model. The remaining samples as the independent test set was used to compare the sample's true age and model estimated age, and had an accuracy rate in the statistical error range of ±0.8 and ±1.0 years, respectively. Then the mean absolute error （MAE） and root mean square error （RMSE） were calculated. Results The accuracy rate of male in the statistical error range of ±0.8 and ±1.0 year was 80.67%, 89.33%, respectively. The MAE and RMSE were 0.486 and 0.606 years, respectively. The accuracy rate of female in the statistical error range of ±0.8 and ±1.0 years was 80.19%, 90.45%, respectively. The MAE and RMSE were 0.485 and 0.590 years, respectively. Conclusion Establishment of prediction model for bone age estimation by feature dimension reduction of HOG and LBP in DR images of knee-joint based on PCA and SVM has relatively high accuracy.   题目:  基于主成分分析和支持向量机实现膝关节骨龄评估回归算法.   摘要:  目的 通过方向梯度直方图（histogram of oriented gradient，HOG）、局部二值模式（local binary patterns，LBP）、支持向量机（support vector machine，SVM）以及主成分分析（principal component analysis，PCA）等机器学习方法构建适用于我国维吾尔族青少年骨龄评估的回归算法模型。 方法 采集维吾尔族12.0~&lt;19.0岁青少年的膝关节DR摄片图像，其中男性样本275例、女性样本225例，采用PCA法对提取的HOG与LBP特征图像进行降维，再以支持向量回归（support vector regression，SVR）算法构建膝关节骨龄评估算法模型。采用随机分层抽样法分别选取男性样本215例、女性样本180例作为SVR模型训练集，并用K折交叉验证法优化模型参数。剩余样本作为独立测试集，将模型预报年龄与样本真实年龄相比，统计误差范围分别在±0.8岁、±1.0岁的准确率，同时计算平均绝对误差（mean absolute error，MAE）与均方根误差（root mean square error，RMSE）。 结果 男性年龄误差范围在±0.8岁及±1.0岁的准确率分别为80.67%和89.33%，MAE为0.486岁，RMSE为0.606岁；女性年龄误差范围在±0.8岁及±1.0岁的准确率分别为80.19%和90.45%，MAE为0.485岁，RMSE为0.590岁。 结论 基于PCA与SVM对膝关节DR摄片图像HOG及LBP特征降维建立骨龄的预报模型，具有较高的准确性。.   关键词:  法医人类学；年龄测定，骨骼；膝关节；支持向量机；主成分分析；方向梯度直方图；局部二值模式；维吾尔族；青少年. \"^^xsd:string ;\n",
      "    ns8:name \"Y Y Lei; Y S Shen; Y H Wang; H Zhao; \"^^xsd:string ;\n",
      "    ns5:journal \"Fa Yi Xue Za Zhi\"^^xsd:string ;\n",
      "    ns7:key_abs \"  forensic anthropology; age determination by skeleton; knee joint; support vector machine; principal component analysis; histogram of oriented gradient; local binary patterns; Uygur nationality; adolescent; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  forensic anthropology; age determination by skeleton; knee joint; support vector machine; principal component analysis; histogram of oriented gradient; local binary patterns; Uygur nationality; adolescent;   Adolescent;  Adult;  Age Determination by Skeleton / methods*;  Algorithms*;  Asian Continental Ancestry Group / ethnology;  China;  Female;  Humans;  Image Processing, Computer-Assisted;  Knee Joint / diagnostic imaging*;  Machine Learning;  Male;  Principal Component Analysis;  Support Vector Machine*;  Young Adult;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adolescent;  Adult;  Age Determination by Skeleton / methods*;  Algorithms*;  Asian Continental Ancestry Group / ethnology;  China;  Female;  Humans;  Image Processing, Computer-Assisted;  Knee Joint / diagnostic imaging*;  Machine Learning;  Male;  Principal Component Analysis;  Support Vector Machine*;  Young Adult;  \"^^xsd:string ;\n",
      "    ns2:Title 31135114 ;\n",
      "    ns4:volume \"35\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.12116/j.issn.1004-5619.2019.02.012']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29577701,27404614,29404850,31145684,32250086\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31136171> a 31136171 ;\n",
      "    ns9:abstract \" An unambiguous algorithm, added to the study of the applicability domain and appropriate measures of the goodness of fit and robustness, represent the key characteristics that should be ideally fulfilled for a QSAR model to be considered for regulatory purposes. In this paper, we propose a new algorithm (RINH) based on the rivality index for the construction of QSAR classification models. This index is capable of predicting the activity of the data set molecules by means of a measurement of the rivality between their nearest neighbors belonging to different classes, contributing with a robust measurement of the reliability of the predictions. In order to demonstrate the goodness of the proposed algorithm we have selected four independent and orthogonally different benchmark data sets (balanced/unbalanced and high/low modelable) and we have compared the results with those obtained using 12 different machine learning algorithms. These results have been validated using 20 data sets of different balancing and sizes, corroborating that the proposed algorithm is able to generate highly accurate classification models and contribute with valuable measurements of the reliability of the predictions and the applicability domain of the built models. \"^^xsd:string ;\n",
      "    ns8:name \"Irene Luque Ruiz; Miguel Ángel Gómez-Nieto; \"^^xsd:string ;\n",
      "    ns5:journal \"J Chem Inf Model\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Algorithms*;  Drug Discovery*;  Machine Learning;  Models, Molecular;  Molecular Conformation;  Quantitative Structure-Activity Relationship*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms*;  Drug Discovery*;  Machine Learning;  Models, Molecular;  Molecular Conformation;  Quantitative Structure-Activity Relationship*;  \"^^xsd:string ;\n",
      "    ns2:Title 31136171 ;\n",
      "    ns4:volume \"59\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1021/acs.jcim.9b00264']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31469296,30356020,30149700,19548346,30372648,32183141\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31145684> a 31145684 ;\n",
      "    ns9:abstract \" Due to the excitability of neurons in the brain, a neurological disorder is produced known as epilepsy. The brain activity of patients suffering from epilepsy is monitored through electroencephalography (EEG). The multivariate nature of features from time domain, frequency domain, complexity and wavelet entropy based, and the statistical features were extracted from healthy and epileptic subjects using the Bonn University database and seizure and non-seizure intervals using the CHB MIT database. The robust machine learning regression methods based on regression, support vector regression (SVR), regression tree (RT), ensemble regression, Gaussian process regression (GPR) were employed for detecting and predicting epileptic seizures. Performance was measured in terms of root mean square error (RMSE), squared error, mean square error (MSE) and mean absolute error (MAE). Moreover, detailed optimization was performed using a RT to predict the selected features from each feature category. A deeper analysis was conducted on features and tree regression methods where optimal RMSE and MSE results were obtained. The best optimal performance was obtained using the ensemble boosted regression tree (BRT) and exponential GPR with an RMSE of 0.47, an MSE (0.22), an R Square (RS) (0.25) and an MAE (0.30) using the Bonn University database and support vector machine (SVM) fine Gaussian with RMSE (0.63634), RS (0.03), MSE (0.40493) and MAE (0.31744); squared exponential GPR and rational quadratic GPR with an RMSE of 0.63841, an RS (0.03), an MSE (0.40757) and an MAE (0.3472) was obtained using the CHB MIT database. A further deeper analysis for the prediction of selected features was performed on an RT to compute the optimal feasible point, observed and estimated function values, function evaluation time, objective function evaluation time and overall elapsed time. \"^^xsd:string ;\n",
      "    ns8:name \"Lal Hussain; Sharjil Saeed; Adnan Idris; Imtiaz Ahmed Awan; Saeed Arif Shah; Abdul Majid; Bilal Ahmed; Quratul-Ain Chaudhary; \"^^xsd:string ;\n",
      "    ns5:journal \"Biomed Tech (Berl)\"^^xsd:string ;\n",
      "    ns7:key_abs \"  K-nearest neighbors; classification; decision tree; ensemble classifier; epilepsy; seizure detection; support vector machine; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  K-nearest neighbors; classification; decision tree; ensemble classifier; epilepsy; seizure detection; support vector machine;   Review;  Brain / physiopathology*;  Databases, Factual;  Electroencephalography / methods*;  Entropy;  Epilepsy / physiopathology*;  Humans;  Machine Learning;  Regression Analysis;  Seizures / physiopathology*;  Support Vector Machine;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Review;  Brain / physiopathology*;  Databases, Factual;  Electroencephalography / methods*;  Entropy;  Epilepsy / physiopathology*;  Humans;  Machine Learning;  Regression Analysis;  Seizures / physiopathology*;  Support Vector Machine;  \"^^xsd:string ;\n",
      "    ns2:Title 31145684 ;\n",
      "    ns4:volume \"64\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.degruyter.com/doi/10.1515/bmt-2018-0012']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29765477,31154257,25614384,31478577,32451639\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31278668> a 31278668 ;\n",
      "    ns9:abstract \" In the post genomic era, large and complex molecular datasets from genome and metagenome sequencing projects expand the limits of what is possible for bioinformatic analyses. Network-based methods are increasingly used to complement phylogenetic analysis in studies in molecular evolution, including comparative genomics, classification, and ecological studies. Using network methods, the vertical and horizontal relationships between all genes or genomes, whether they are from cellular chromosomes or mobile genetic elements, can be explored in a single expandable graph. In recent years, development of new methods for the construction and analysis of networks has helped to broaden the availability of these approaches from programmers to a diversity of users. This chapter introduces the different kinds of networks based on sequence similarity that are already available to tackle a wide range of biological questions, including sequence similarity networks, gene-sharing networks and bipartite graphs, and a guide for their construction and analyses. \"^^xsd:string ;\n",
      "    ns8:name \"Andrew K Watson; Romain Lannes; Jananan S Pathmanathan; Raphaël Méheust; Slim Karkar; Philippe Colson; Eduardo Corel; Philippe Lopez; Eric Bapteste; \"^^xsd:string ;\n",
      "    ns5:journal \"Methods Mol Biol\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Ecology; Evolution; Gene remodeling; Lateral gene transfer (LGT); Metagenomics; Sequence similarity network; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Ecology; Evolution; Gene remodeling; Lateral gene transfer (LGT); Metagenomics; Sequence similarity network;   Biodiversity;  Biological Evolution;  Computational Biology / methods;  Ecosystem;  Evolution, Molecular;  Gene Ontology;  Gene Regulatory Networks;  High-Throughput Nucleotide Sequencing;  Metagenome*;  Metagenomics* / methods;  Microbiota;  Molecular Sequence Annotation;  Multigene Family;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Biodiversity;  Biological Evolution;  Computational Biology / methods;  Ecosystem;  Evolution, Molecular;  Gene Ontology;  Gene Regulatory Networks;  High-Throughput Nucleotide Sequencing;  Metagenome*;  Metagenomics* / methods;  Microbiota;  Molecular Sequence Annotation;  Multigene Family;  \"^^xsd:string ;\n",
      "    ns2:Title 31278668 ;\n",
      "    ns4:volume \"1910\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/978-1-4939-9074-0_9']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29651978,26774999,30287354,31278679,27899624\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31285275> a 31285275 ;\n",
      "    ns9:abstract \" Navigating through natural environments requires localizing objects along three distinct spatial axes. Information about position along the horizontal and vertical axes is available from an object's position on the retina, while position along the depth axis must be inferred based on second-order cues such as the disparity between the images cast on the two retinae. Past work has revealed that object position in two-dimensional (2D) retinotopic space is robustly represented in visual cortex and can be robustly predicted using a multivariate encoding model, in which an explicit axis is modeled for each spatial dimension. However, no study to date has used an encoding model to estimate a representation of stimulus position in depth. Here, we recorded BOLD fMRI while human subjects viewed a stereoscopic random-dot sphere at various positions along the depth (<i>z</i>) and the horizontal (<i>x</i>) axes, and the stimuli were presented across a wider range of disparities (out to ∼40 arcmin) compared to previous neuroimaging studies. In addition to performing decoding analyses for comparison to previous work, we built encoding models for depth position and for horizontal position, allowing us to directly compare encoding between these dimensions. Our results validate this method of recovering depth representations from retinotopic cortex. Furthermore, we find convergent evidence that depth is encoded most strongly in dorsal area V3A. \"^^xsd:string ;\n",
      "    ns8:name \"Margaret Henderson; Vy Vo; Chaipat Chunharas; Thomas Sprague; John Serences; \"^^xsd:string ;\n",
      "    ns5:journal \"eNeuro\"^^xsd:string ;\n",
      "    ns7:key_abs \"  MVPA; depth; encoding model; fMRI; intraparietal sulcus; vision; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  MVPA; depth; encoding model; fMRI; intraparietal sulcus; vision;   Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Brain Mapping;  Depth Perception / physiology*;  Female;  Humans;  Magnetic Resonance Imaging;  Male;  Models, Neurological;  Multivariate Analysis;  Parietal Lobe / physiology*;  Photic Stimulation / methods;  Support Vector Machine;  Visual Cortex / physiology*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Brain Mapping;  Depth Perception / physiology*;  Female;  Humans;  Magnetic Resonance Imaging;  Male;  Models, Neurological;  Multivariate Analysis;  Parietal Lobe / physiology*;  Photic Stimulation / methods;  Support Vector Machine;  Visual Cortex / physiology*;  \"^^xsd:string ;\n",
      "    ns2:Title 31285275 ;\n",
      "    ns4:volume \"6\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://eneuro.org/cgi/pmidlookup?view=long&pmid=31285275', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31285275/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"18971473,26290233,28039760,10605650,7613075,31836597\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31317398> a 31317398 ;\n",
      "    ns9:abstract \"  Objective:  Artificial neural network (ANN) technology has been developed for clinical use to analyze bone scintigraphy with metastatic bone tumors. It has been reported to improve diagnostic accuracy and reproducibility especially in cases of prostate cancer. The aim of this study was to evaluate the diagnostic usefulness of quantitative bone scintigraphy with ANN in patients having breast cancer.   Patients and methods:  We retrospectively evaluated 88 patients having breast cancer who underwent both bone scintigraphy and <sup>18</sup>F-fluorodeoxyglucose (FDG) positron-emission computed tomography/X-ray computed tomography (PET/CT) within an interval of 8 weeks between both examinations for comparison. The whole-body bone images were analyzed with fully automated software that was customized according to a Japanese multicenter database. The region of interest for FDG-PET was set to bone lesions in patients with bone metastasis, while the bone marrow of the ilium and the vertebra was used in patients without bone metastasis.   Results:  Thirty of 88 patients had bone metastasis. Extent of disease, bone scan index (BSI) which indicate severity of bone metastasis, the maximum standardized uptake value (SUVmax), metabolic tumor volume (MTV), total lesion glycolysis (TLG), and serum tumor markers in patients with bone metastasis were significantly higher than those in patients without metastasis. The Kaplan-Meier survival curve showed that the overall survival of the lower BSI group was longer than that with the higher BSI group in patients with visceral metastasis. In the multivariate Cox proportional hazard model, BSI (hazard ratio (HR): 19.15, p = 0.0077) and SUVmax (HR: 10.12, p = 0.0068) were prognostic factors in patients without visceral metastasis, while the BSI was only a prognostic factor in patients with visceral metastasis (HR: 7.88, p = 0.0084), when dividing the sample into two groups with each mean value in patients with bone metastasis.   Conclusion:  BSI, an easily and automatically calculated parameter, was a well prognostic factor in patients with visceral metastasis as well as without visceral metastasis from breast cancer. \"^^xsd:string ;\n",
      "    ns8:name \"Anri Inaki; Kenichi Nakajima; Hiroshi Wakabayashi; Takafumi Mochizuki; Seigo Kinuya; \"^^xsd:string ;\n",
      "    ns5:journal \"Ann Nucl Med\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Artificial neural network; Bone scan; Breast cancer with bone metastasis; FDG-PET; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Artificial neural network; Bone scan; Breast cancer with bone metastasis; FDG-PET;   Automation;  Bone Neoplasms / diagnostic imaging;  Bone Neoplasms / secondary;  Bone and Bones / diagnostic imaging*;  Breast Neoplasms / diagnostic imaging*;  Breast Neoplasms / pathology;  Female;  Fluorodeoxyglucose F18;  Humans;  Image Processing, Computer-Assisted / methods*;  Middle Aged;  Neural Networks, Computer*;  Positron Emission Tomography Computed Tomography*;  Retrospective Studies;  Whole Body Imaging;  Fluorodeoxyglucose F18;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Automation;  Bone Neoplasms / diagnostic imaging;  Bone Neoplasms / secondary;  Bone and Bones / diagnostic imaging*;  Breast Neoplasms / diagnostic imaging*;  Breast Neoplasms / pathology;  Female;  Fluorodeoxyglucose F18;  Humans;  Image Processing, Computer-Assisted / methods*;  Middle Aged;  Neural Networks, Computer*;  Positron Emission Tomography Computed Tomography*;  Retrospective Studies;  Whole Body Imaging;  Fluorodeoxyglucose F18;  \"^^xsd:string ;\n",
      "    ns2:Title 31317398 ;\n",
      "    ns4:volume \"33\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/s12149-019-01386-1']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30843003,24403238,26204115,22999369,31036925\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31320356> a 31320356 ;\n",
      "    ns9:abstract \"  Introduction:  Coma is a deep state of unconsciousness that can be caused by a variety of clinical conditions. Traditional tests for coma outcome prediction are based mainly on a set of clinical observations. Recently, certain event-related potentials (ERPs), which are transient electroencephalogram (EEG) responses to auditory, visual or tactile stimuli, have been introduced as useful predictors of a positive coma outcome (ie, emergence). However, such tests require the skills of clinical neurophysiologists, who are not commonly available in many clinical settings. Additionally, none of the current standard clinical approaches have sufficient predictive accuracies to provide definitive prognoses.   Objective:  The objective of this study is to develop improved machine learning procedures based on EEG/ERP for determining emergence from coma.   Methods and analysis:  Data will be collected from 50 participants in coma. EEG/ERP data will be recorded for 24 consecutive hours at a maximum of five time points spanning 30 days from the date of recruitment to track participants' progression. The study employs paradigms designed to elicit brainstem potentials, middle-latency responses, N100, mismatch negativity, P300 and N400. In the case of patient emergence, data are recorded on that occasion to form an additional basis for comparison. A relevant data set will be developed from the testing of 20 healthy controls, each spanning a 15-hour recording period in order to formulate a baseline. Collected data will be used to develop an automated procedure for analysis and detection of various ERP components that are salient to prognosis. Salient features extracted from the ERP and resting-state EEG will be identified and combined to give an accurate indicator of prognosis.   Ethics and dissemination:  This study is approved by the Hamilton Integrated Research Ethics Board (project number 4840). Results will be disseminated through peer-reviewed journal articles and presentations at scientific conferences.   Trial registration number:  <a href=\\\"http://clinicaltrials.gov/show/NCT03826407\\\" title=\\\"See in ClinicalTrials.gov\\\">NCT03826407</a>. \"^^xsd:string ;\n",
      "    ns8:name \"John F Connolly; James P Reilly; Alison Fox-Robichaud; Patrick Britz; Stefanie Blain-Moraes; Ranil Sonnadara; Cindy Hamielec; Adianes Herrera-Díaz; Rober Boshra; \"^^xsd:string ;\n",
      "    ns5:journal \"BMJ Open\"^^xsd:string ;\n",
      "    ns7:key_abs \"  neurological injury; neurophysiology; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  neurological injury; neurophysiology;   Observational Study;  Research Support, Non-U.S. Gov't;  Brain / physiopathology*;  Coma / diagnosis*;  Coma / pathology;  Electroencephalography;  Evoked Potentials;  Humans;  Machine Learning;  Point-of-Care Systems*;  Prognosis;  Prospective Studies;  Research Design;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Observational Study;  Research Support, Non-U.S. Gov't;  Brain / physiopathology*;  Coma / diagnosis*;  Coma / pathology;  Electroencephalography;  Evoked Potentials;  Humans;  Machine Learning;  Point-of-Care Systems*;  Prognosis;  Prospective Studies;  Research Design;  \"^^xsd:string ;\n",
      "    ns2:Title 31320356 ;\n",
      "    ns4:volume \"9\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://bmjopen.bmj.com/cgi/pmidlookup?view=long&pmid=31320356', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31320356/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"8857436,8612421,23165873,25701685,29784540,31758044\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31333196> a 31333196 ;\n",
      "    ns9:abstract \"  Background:  As the most commonly occurring form of mental illness worldwide, depression poses significant health and economic burdens to both the individual and community. Different types of depression pose different levels of risk. Individuals who suffer from mild forms of depression may recover without any assistance or be effectively managed by primary care or family practitioners. However, other forms of depression are far more severe and require advanced care by certified mental health providers. However, identifying cases of depression that require advanced care may be challenging to primary care providers and health care team members whose skill sets run broad rather than deep.   Objective:  This study aimed to leverage a comprehensive range of patient-level diagnostic, behavioral, and demographic data, as well as past visit history data from a statewide health information exchange to build decision models capable of predicting the need of advanced care for depression across patients presenting at Eskenazi Health, the public safety net health system for Marion County, Indianapolis, Indiana.   Methods:  Patient-level diagnostic, behavioral, demographic, and past visit history data extracted from structured datasets were merged with outcome variables extracted from unstructured free-text datasets and were used to train random forest decision models that predicted the need of advanced care for depression across (1) the overall patient population and (2) various subsets of patients at higher risk for depression-related adverse events; patients with a past diagnosis of depression; patients with a Charlson comorbidity index of ≥1; patients with a Charlson comorbidity index of ≥2; and all unique patients identified across the 3 above-mentioned high-risk groups.   Results:  The overall patient population consisted of 84,317 adult (aged ≥18 years) patients. A total of 6992 (8.29%) of these patients were in need of advanced care for depression. Decision models for high-risk patient groups yielded area under the curve (AUC) scores between 86.31% and 94.43%. The decision model for the overall patient population yielded a comparatively lower AUC score of 78.87%. The variance of optimal sensitivity and specificity for all decision models, as identified using Youden J Index, is as follows: sensitivity=68.79% to 83.91% and specificity=76.03% to 92.18%.   Conclusions:  This study demonstrates the ability to automate screening for patients in need of advanced care for depression across (1) an overall patient population or (2) various high-risk patient groups using structured datasets covering acute and chronic conditions, patient demographics, behaviors, and past visit history. Furthermore, these results show considerable potential to enable preventative care and can be easily integrated into existing clinical workflows to improve access to wraparound health care services. \"^^xsd:string ;\n",
      "    ns8:name \"Suranga N Kasthurirathne; Paul G Biondich; Shaun J Grannis; Saptarshi Purkayastha; Joshua R Vest; Josette F Jones; \"^^xsd:string ;\n",
      "    ns5:journal \"J Med Internet Res\"^^xsd:string ;\n",
      "    ns7:key_abs \"  delivery of health care; depression; supervised machine learning; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  delivery of health care; depression; supervised machine learning;   Adolescent;  Adult;  Delivery of Health Care / methods*;  Depression / therapy*;  Female;  Health Information Exchange / standards*;  Humans;  Machine Learning / standards*;  Male;  Middle Aged;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Adolescent;  Adult;  Delivery of Health Care / methods*;  Depression / therapy*;  Female;  Health Information Exchange / standards*;  Humans;  Machine Learning / standards*;  Male;  Middle Aged;  \"^^xsd:string ;\n",
      "    ns2:Title 31333196 ;\n",
      "    ns4:volume \"21\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.jmir.org/2019/7/e13809/', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31333196/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"32479414,27606392,29177457,31679171,28876770,32715427,32477634\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31374758> a 31374758 ;\n",
      "    ns9:abstract \" Clinical viability of powered lower-limb assistive devices requires reliable and intuitive control strategies. Stance and swing are the main phases of the gait cycle across different locomotion tasks. Hence, a reliable method to accurately identify these phases can decrease sensing complexity and assist in enabling high-level control of assistive devices. Ultrasound (US) imaging has recently been introduced as a new sensing modality that may provide a solution for intuitive device control. US images of the rectus femoris and vastus intermedius muscles were collected in humans during level, incline, and decline ambulation tasks. Five low-level static (i.e. time-independent) features of US images were measured with respect to a reference image, including correlation coefficient, sum of absolute differences, structural similarity index, sum of squared differences, and image echogenicity. Time-derivatives of the static features were also calculated as temporal features. Support vector machine classifiers were trained using these static features to identify the gait phase both dependent and independent of the ambulation tasks. The results indicate an accuracy of 88.3% in identifying the gait phases for task-independent classifiers when trained using only the static features. Performance of the classifiers improved significantly to 92.8% after using the temporal features (p $\\\\lt0.01)$. The algorithm was efficient and the average processing speed was faster than 100 Hz. This study is the first demonstration on use of US imaging to provide continuous estimates of ambulation phase, and on multiple surfaces. These findings suggest task-independent approaches may reliably identify the main phases of the gait cycle. Advancements in this area of study may provide simpler intuitive strategies for high-level assistive device control and increase their clinical relevance. \"^^xsd:string ;\n",
      "    ns8:name \"Mohammad Hassan Jahanandish; Kaitlin G Rabe; Nicholas P Fey; Kenneth Hoyt; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Int Conf Rehabil Robot\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, N.I.H., Extramural;  Adult;  Algorithms;  Female;  Gait / physiology*;  Humans;  Locomotion / physiology;  Lower Extremity / physiology*;  Male;  Muscle, Skeletal / physiology;  Support Vector Machine;  Walking / physiology;  Young Adult;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Adult;  Algorithms;  Female;  Gait / physiology*;  Humans;  Locomotion / physiology;  Lower Extremity / physiology*;  Male;  Muscle, Skeletal / physiology;  Support Vector Machine;  Walking / physiology;  Young Adult;  \"^^xsd:string ;\n",
      "    ns2:Title 31374758 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/ICORR.2019.8779534']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30629522,27362983,28422559,29433090,24814597\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31374796> a 31374796 ;\n",
      "    ns9:abstract \" In a stable bimanual trajectory tracing task with interlimb spatial and temporal synchrony, blocking the visual information from one hand may alter the performance of either hand. In this paper, we investigate the effect of visual information on motor behaviour of dominant and non-dominant hands during a bimanual task, with a focus on motor lateralization theory's anticipation for a more pronounced distortion on one hand due to visual information withdrawal. To address this question, four bimanual circle tracing experiments were designed with two rehabilitation robotic arms with real time visual feedback. Two experiments were conducted under the free-visual condition whereas the visual feedback from one hand was blocked for the other two. The in-depth analysis of the metrics extracted from 685 circles, drawn by 6 participants, revealed that non-dominant hand, when visible, generally performs worse than the dominant hand, for instance it exhibits less circularity. In their invisible modes, the performance of the dominant and non-dominant hands displayed inconsistent difference across the participants. Moreover, both hands showed a higher pace when partial visual information was available. Our findings using this robotic framework as a systematic tool on developing new paradigms are discussed. \"^^xsd:string ;\n",
      "    ns8:name \"Mina Nouredanesh; Matthew Frazer; James Tung; Soo Jeon; Arash Arami; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Int Conf Rehabil Robot\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Adult;  Feedback, Sensory / physiology;  Female;  Hand / physiopathology*;  Humans;  Male;  Psychomotor Performance / physiology;  Robotic Surgical Procedures / methods*;  Robotics*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Adult;  Feedback, Sensory / physiology;  Female;  Hand / physiopathology*;  Humans;  Male;  Psychomotor Performance / physiology;  Robotic Surgical Procedures / methods*;  Robotics*;  \"^^xsd:string ;\n",
      "    ns2:Title 31374796 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/ICORR.2019.8779461']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"27655023,10442410,26779835,24215820,23943555\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31399493> a 31399493 ;\n",
      "    ns9:abstract \" Social recognition in mammals depends on complex interactions between sensory and other brain areas as well as modulatory inputs by specific neuropeptides such as oxytocin (OXT). Social recognition memory specifically has been shown to depend among others on olfactory processing, and can be probed using methods similar to those used when probing non-social odor memory. We here use a computational model of two interconnected olfactory networks in the mouse, the olfactory bulb (OB) and anterior olfactory nucleus, to propose a mechanism for olfactory short-term recognition memory and its modulation in social situations. Based on previous experiments, we propose one early locus for memory to be the OB. During social encounters in mice, pyramidal cells in the anterior olfactory nucleus, themselves driven by olfactory input, are rendered more excitable by OXT release, resulting in stronger feedback to OB local interneurons. This additional input to the OB creates stronger dynamics and improves signal-to-noise ratio of odor responses in the OB proper. As a consequence, mouse social olfactory memories are more strongly encoded and their duration is modulated. \"^^xsd:string ;\n",
      "    ns8:name \"Christiane Linster; Wolfgang Kelsch; \"^^xsd:string ;\n",
      "    ns5:journal \"eNeuro\"^^xsd:string ;\n",
      "    ns7:key_abs \"  anterior olfactory nucleus; computation; olfaction; oxytocin; plasticity; social odors; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  anterior olfactory nucleus; computation; olfaction; oxytocin; plasticity; social odors;   Research Support, Non-U.S. Gov't;  Animals;  Female;  Male;  Mice, Inbred C57BL;  Models, Neurological*;  Neural Networks, Computer;  Neural Pathways / physiology;  Neurons / physiology*;  Olfactory Bulb / physiology*;  Olfactory Cortex / physiology*;  Oxytocin / physiology*;  Recognition, Psychology / physiology*;  Social Behavior;  Oxytocin;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Animals;  Female;  Male;  Mice, Inbred C57BL;  Models, Neurological*;  Neural Networks, Computer;  Neural Pathways / physiology;  Neurons / physiology*;  Olfactory Bulb / physiology*;  Olfactory Cortex / physiology*;  Oxytocin / physiology*;  Recognition, Psychology / physiology*;  Social Behavior;  Oxytocin;  \"^^xsd:string ;\n",
      "    ns2:Title 31399493 ;\n",
      "    ns4:volume \"6\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://eneuro.org/cgi/pmidlookup?view=long&pmid=31399493', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31399493/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"27112498,27511004,21920364,10888874,26042330\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31400112> a 31400112 ;\n",
      "    ns9:abstract \" Natural products represent a rich reservoir of small molecule drug candidates utilized as antimicrobial drugs, anticancer therapies, and immunomodulatory agents. These molecules are microbial secondary metabolites synthesized by co-localized genes termed Biosynthetic Gene Clusters (BGCs). The increase in full microbial genomes and similar resources has led to development of BGC prediction algorithms, although their precision and ability to identify novel BGC classes could be improved. Here we present a deep learning strategy (DeepBGC) that offers reduced false positive rates in BGC identification and an improved ability to extrapolate and identify novel BGC classes compared to existing machine-learning tools. We supplemented this with random forest classifiers that accurately predicted BGC product classes and potential chemical activity. Application of DeepBGC to bacterial genomes uncovered previously undetectable putative BGCs that may code for natural products with novel biologic activities. The improved accuracy and classification ability of DeepBGC represents a major addition to in-silico BGC identification. \"^^xsd:string ;\n",
      "    ns8:name \"Geoffrey D Hannigan; David Prihoda; Andrej Palicka; Jindrich Soukup; Ondrej Klempir; Lena Rampula; Jindrich Durcak; Michael Wurst; Jakub Kotowski; Dan Chang; Rurun Wang; Grazia Piizzi; Gergely Temesi; Daria J Hazuda; Christopher H Woelk; Danny A Bitton; \"^^xsd:string ;\n",
      "    ns5:journal \"Nucleic Acids Res\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Biosynthetic Pathways / genetics*;  Computational Biology / methods*;  Data Mining / methods*;  Deep Learning;  Genome;  Genome, Bacterial / genetics;  Multigene Family / genetics*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Biosynthetic Pathways / genetics*;  Computational Biology / methods*;  Data Mining / methods*;  Deep Learning;  Genome;  Genome, Bacterial / genetics;  Multigene Family / genetics*;  \"^^xsd:string ;\n",
      "    ns2:Title 31400112 ;\n",
      "    ns4:volume \"47\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkz654', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31400112/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29112695,30992349,26458099,28587290,30859257,32637051,32427317\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31416053> a 31416053 ;\n",
      "    ns9:abstract \" We propose an ensemble of multilayer feedforward neural networks to estimate the 3D position of photoelectric interactions in monolithic detectors. The ensemble is trained with data generated from optical Monte Carlo simulations only. The originality of our approach is to exploit simulations to obtain reference data, in combination with a variability reduction that the network ensembles offer, thus, removing the need of extensive per-detector calibration measurements. This procedure delivers an ensemble valid for any detector of the same design. We show the capability of the ensemble to solve the 3D positioning problem through testing four different detector designs with Monte Carlo data, measurements from physical detectors and reconstructed images from the MindView scanner. Network ensembles allow the detector to achieve a 2-2.4 mm FWHM, depending on its design, and the associated reconstructed images present improved SNR, CNR and SSIM when compared to those based on the MindView built-in positioning algorithm. \"^^xsd:string ;\n",
      "    ns8:name \"A Iborra; A J González; A González-Montoro; A Bousse; D Visvikis; \"^^xsd:string ;\n",
      "    ns5:journal \"Phys Med Biol\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Algorithms;  Brain / diagnostic imaging*;  Calibration;  Computer Simulation;  Humans;  Image Processing, Computer-Assisted / methods*;  Imaging, Three-Dimensional;  Light;  Models, Statistical;  Monte Carlo Method;  Neural Networks, Computer*;  Optics and Photonics;  Phantoms, Imaging;  Positron-Emission Tomography*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Brain / diagnostic imaging*;  Calibration;  Computer Simulation;  Humans;  Image Processing, Computer-Assisted / methods*;  Imaging, Three-Dimensional;  Light;  Models, Statistical;  Monte Carlo Method;  Neural Networks, Computer*;  Optics and Photonics;  Phantoms, Imaging;  Positron-Emission Tomography*;  \"^^xsd:string ;\n",
      "    ns2:Title 31416053 ;\n",
      "    ns4:volume \"64\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1088/1361-6560/ab3b86']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30667069,26305868,31292287,17664575,23566478\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31416552> a 31416552 ;\n",
      "    ns9:abstract \"  Background and objective:  Retinal fundus images are widely used to diagnose retinal diseases and can potentially be used for early diagnosis and prevention of chronic vascular diseases and diabetes. While various automatic retinal vessel segmentation methods using deep learning have been proposed, they are mostly based on common CNN structures developed for other tasks such as classification.   Methods:  We present a novel and simple multi-scale convolutional neural network (CNN) structure for retinal vessel segmentation. We first provide a theoretical analysis of existing multi-scale structures based on signal processing. In previous structures, multi-scale representations are achieved through downsampling by subsampling and decimation. By incorporating scale-space theory, we propose a simple yet effective multi-scale structure for CNNs using upsampling, which we term scale-space approximated CNN (SSANet). Based on further analysis of the effects of the SSA structure within a CNN, we also incorporate residual blocks, resulting in a multi-scale CNN that outperforms current state-of-the-art methods.   Results:  Quantitative evaluations are presented as the area-under-curve (AUC) of the receiver operating characteristic (ROC) curve and the precision-recall curve, as well as accuracy, for four publicly available datasets, namely DRIVE, STARE, CHASE_DB1, and HRF. For the CHASE_DB1 set, the SSANet achieves state-of-the-art AUC value of 0.9916 for the ROC curve. An ablative analysis is presented to analyze the contribution of different components of the SSANet to the performance improvement.   Conclusions:  The proposed retinal SSANet achieves state-of-the-art or comparable accuracy across publicly available datasets, especially improving segmentation for thin vessels, vessel junctions, and central vessel reflexes. \"^^xsd:string ;\n",
      "    ns8:name \"Kyoung Jin Noh; Sang Jun Park; Soochahn Lee; \"^^xsd:string ;\n",
      "    ns5:journal \"Comput Methods Programs Biomed\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Convolutional neural networks; Multi-scale representation; Retinal vessel segmentation; Scale-space approximation; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Convolutional neural networks; Multi-scale representation; Retinal vessel segmentation; Scale-space approximation;   Algorithms;  Area Under Curve;  Deep Learning;  False Positive Reactions;  Fundus Oculi;  Humans;  Image Processing, Computer-Assisted;  Neural Networks, Computer*;  Normal Distribution;  ROC Curve;  Retinal Diseases / diagnostic imaging*;  Retinal Vessels / diagnostic imaging*;  Signal Processing, Computer-Assisted;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms;  Area Under Curve;  Deep Learning;  False Positive Reactions;  Fundus Oculi;  Humans;  Image Processing, Computer-Assisted;  Neural Networks, Computer*;  Normal Distribution;  ROC Curve;  Retinal Diseases / diagnostic imaging*;  Retinal Vessels / diagnostic imaging*;  Signal Processing, Computer-Assisted;  \"^^xsd:string ;\n",
      "    ns2:Title 31416552 ;\n",
      "    ns4:volume \"178\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0169-2607(19)30222-6']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30501859,29775951,30441687,22525589,28474130,32165702\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31422557> a 31422557 ;\n",
      "    ns9:abstract \" Helitrons are mobile sequences which belong to the class 2 of eukaryotic transposons. Their specificity resides in their mechanism of transposition: the rolling circle mechanism. They play an important role in remodeling proteomes due to their ability to modify existing genes and introducing new ones. A major difficulty in identifying and classifying Helitron families comes from the complex structure, the unspecified length, and the unbalanced appearance number of each Helitron type. The Helitron's recognition is still not solved in literature. The purpose of this paper is to characterize and classify Helitron types using spectral features and support vector machine (SVM) classification technique. Thus, the helitronic DNA is transformed into a numerical form using the FCGS<sub>2</sub> coding technique. Then, a set of spectral features is extracted from the smoothed Fourier transform applied on the FCGS<sub>2</sub> signals. Based on the spectral signature and the classification's confusion matrix, we demonstrated that some specific classes which do not show similarities, such as HelitronY2 and NDNAX3, are easily discriminated with important accuracy rates exceeding 90%. However, some Helitron types have great similarities such as the following: Helitron1, HelitronY1, HelitronY1A, and HelitronY4. Our system is also able to predict them with promising values reaching 70%. Graphical abstract The Helitron recognizer based on features extracted from smoothed Fourier transform. \"^^xsd:string ;\n",
      "    ns8:name \"Rabeb Touati; Afef Elloumi Oueslati; Imen Messaoudi; Zied Lachiri; \"^^xsd:string ;\n",
      "    ns5:journal \"Med Biol Eng Comput\"^^xsd:string ;\n",
      "    ns7:key_abs \"  C. elegans; FCGS2 coding; Helitrons; SVM classification; Smoothed Fourier transform; Spectral signature; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  C. elegans; FCGS2 coding; Helitrons; SVM classification; Smoothed Fourier transform; Spectral signature;   Animals;  Caenorhabditis elegans / genetics;  Chromosomes / genetics;  Databases as Topic*;  Fourier Analysis*;  Genome, Helminth;  Support Vector Machine*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Animals;  Caenorhabditis elegans / genetics;  Chromosomes / genetics;  Databases as Topic*;  Fourier Analysis*;  Genome, Helminth;  Support Vector Machine*;  \"^^xsd:string ;\n",
      "    ns2:Title 31422557 ;\n",
      "    ns4:volume \"57\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1007/s11517-019-02027-5']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"25480150,17850916,27553634,11447285,18023126,32645523\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31437939> a 31437939 ;\n",
      "    ns9:abstract \"  Objective:  To characterize the scientific reproducibility of biomedical research studies by query and analysis of semantic provenance graphs generated from provenance metadata terms extracted from PubMed articles.   Methods:  We develop a new semantic provenance graph generation algorithm that uses a provenance ontology developed as part of the Provenance for Clinical and Health Research (ProvCaRe) project. The ProvCaRe project has processed and extracted provenance metadata from more than 1.6 million full text articles from the PubMed database.   Results:  The semantic provenance graph generation algorithm is evaluated using provenance terms extracted from 75 selected articles describing sleep medicine research studies. In addition, we use eight provenance queries to evaluate the quality of semantic provenance graphs generated by the new algorithm.   Conclusion:  The ProvCaRe project has created a unique resource to characterize the reproducibility of biomedical research studies and the semantic provenance graph generation algorithm enables users to effectively query and analyze the provenance metadata in the ProvCaRe knowledge repository. \"^^xsd:string ;\n",
      "    ns8:name \"Satya S Sahoo; Joshua Valdez; Michael Rueschman; Matthew Kim; \"^^xsd:string ;\n",
      "    ns5:journal \"Stud Health Technol Inform\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Biomedical Ontologies; Health Information Systems; Knowledge Bases; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Biomedical Ontologies; Health Information Systems; Knowledge Bases;   Biological Ontologies*;  Biomedical Research*;  Metadata;  Reproducibility of Results;  Semantics;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Biological Ontologies*;  Biomedical Research*;  Metadata;  Reproducibility of Results;  Semantics;  \"^^xsd:string ;\n",
      "    ns2:Title 31437939 ;\n",
      "    ns4:volume \"264\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.3233/SHTI190237', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31437939/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30545485,29854241,28269904,28938912,29329592\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31455677> a 31455677 ;\n",
      "    ns9:abstract \" Copy number variants (CNV) are associated with phenotypic variation in several species. However, properly detecting changes in copy numbers of sequences remains a difficult problem, especially in lower quality or lower coverage next-generation sequencing data. Here, inspired by recent applications of machine learning in genomics, we describe a method to detect duplications and deletions in short-read sequencing data. In low coverage data, machine learning appears to be more powerful in the detection of CNVs than the gold-standard methods of coverage estimation alone, and of equal power in high coverage data. We also demonstrate how replicating training sets allows a more precise detection of CNVs, even identifying novel CNVs in two genomes previously surveyed thoroughly for CNVs using long read data. \"^^xsd:string ;\n",
      "    ns8:name \"Tom Hill; Robert L Unckless; \"^^xsd:string ;\n",
      "    ns5:journal \"G3 (Bethesda)\"^^xsd:string ;\n",
      "    ns7:key_abs \"  coverage; deletion; duplication; machine-learning; next-generation sequencing; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  coverage; deletion; duplication; machine-learning; next-generation sequencing;   Research Support, N.I.H., Extramural;  Animals;  DNA Copy Number Variations*;  Deep Learning*;  Drosophila melanogaster / genetics;  Genome, Insect;  Genome, Viral;  Genomics / methods*;  High-Throughput Nucleotide Sequencing*;  Nudiviridae / genetics;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Animals;  DNA Copy Number Variations*;  Deep Learning*;  Drosophila melanogaster / genetics;  Genome, Insect;  Genome, Viral;  Genomics / methods*;  High-Throughput Nucleotide Sequencing*;  Nudiviridae / genetics;  \"^^xsd:string ;\n",
      "    ns2:Title 31455677 ;\n",
      "    ns4:volume \"9\"^^xsd:string ;\n",
      "    rdfs:comment \"['http://www.g3journal.org/cgi/pmidlookup?view=long&pmid=31455677', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31455677/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30343665,27597741,19528327,30221560,24240121\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31466478> a 31466478 ;\n",
      "    ns9:abstract \" <b>Aim:</b> Smoking strongly influences DNA methylation, with current and never smokers exhibiting different methylation profiles. <b>Methods:</b> To advance the practical applicability of the smoking-associated methylation signals, we used machine learning methodology to train a classifier for smoking status prediction. <b>Results:</b> We show the prediction performance of our classifier on three independent whole-blood datasets demonstrating its robustness and global applicability. Furthermore, we examine the reasons for biologically meaningful misclassifications through comprehensive phenotypic evaluation. <b>Conclusion:</b> The major contribution of our classifier is its global applicability without a need for users to determine a threshold value for each dataset to predict the smoking status. We provide an R package, EpiSmokEr (Epigenetic Smoking status Estimator), facilitating the use of our classifier to predict smoking status in future studies. \"^^xsd:string ;\n",
      "    ns8:name \"Sailalitha Bollepalli; Tellervo Korhonen; Jaakko Kaprio; Simon Anders; Miina Ollikainen; \"^^xsd:string ;\n",
      "    ns5:journal \"Epigenomics\"^^xsd:string ;\n",
      "    ns7:key_abs \"  DNA methylation; epigenetic smoking status; multinomial LASSO; smoking status classifier; tobacco smoking; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  DNA methylation; epigenetic smoking status; multinomial LASSO; smoking status classifier; tobacco smoking;   Research Support, Non-U.S. Gov't;  Adult;  Aged;  Computational Biology / methods;  CpG Islands;  DNA Methylation*;  Epigenesis, Genetic;  Epigenomics / methods*;  Female;  Humans;  Machine Learning;  Male;  Middle Aged;  Software;  Tobacco Smoking / genetics*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Adult;  Aged;  Computational Biology / methods;  CpG Islands;  DNA Methylation*;  Epigenesis, Genetic;  Epigenomics / methods*;  Female;  Humans;  Machine Learning;  Male;  Middle Aged;  Software;  Tobacco Smoking / genetics*;  \"^^xsd:string ;\n",
      "    ns2:Title 31466478 ;\n",
      "    ns4:volume \"11\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.futuremedicine.com/doi/10.2217/epi-2019-0206?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub  0pubmed']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29956353,31536415,26864933,30228624,27924599,32484362\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31470662> a 31470662 ;\n",
      "    ns9:abstract \" Processing massive transcriptomic datasets in a meaningful manner requires novel, possibly interdisciplinary, approaches. One principle that can address this challenge is the Benford law (BL), which posits that the occurrence probability of a leading digit in a large numerical dataset decreases as its value increases. Here, we analyzed large single-cell and bulk RNA-seq datasets to test whether cell types and tissue origins can be differentiated based on the adherence of specific genes to the BL. Then, we used the Benford adherence scores of these genes as inputs to machine-learning algorithms and tested their separation accuracy. We found that genes selected based on their first-digit distributions can distinguish between cell types and tissue origins. Moreover, despite the simplicity of this novel feature-selection method, its separation accuracy is higher than that of the mean-expression level approach and is similar to that of the differential expression approach. Thus, the BL can be used to obtain biological insights from massive amounts of numerical genomics data-a capability that could be utilized in various biomedical applications, e.g., to resolve samples of unknown primary origin, identify possible sample contaminations, and provide insights into the molecular basis of cancer subtypes. \"^^xsd:string ;\n",
      "    ns8:name \"Sne Morag; Mali Salmon-Divon; \"^^xsd:string ;\n",
      "    ns5:journal \"Cells\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Benford distribution; Benford law; cell classification; machine learning; single-cell RNA sequencing; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Benford distribution; Benford law; cell classification; machine learning; single-cell RNA sequencing;   Databases, Genetic*;  Datasets as Topic;  Genome, Human / genetics*;  Humans;  Machine Learning;  Probability;  RNA-Seq / statistics & numerical data*;  Single-Cell Analysis / statistics & numerical data*;  Statistical Distributions*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Databases, Genetic*;  Datasets as Topic;  Genome, Human / genetics*;  Humans;  Machine Learning;  Probability;  RNA-Seq / statistics & numerical data*;  Single-Cell Analysis / statistics & numerical data*;  Statistical Distributions*;  \"^^xsd:string ;\n",
      "    ns2:Title 31470662 ;\n",
      "    ns4:volume \"8\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.mdpi.com/resolver?pii=cells8091004', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31470662/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"27506195,30864335,30304022,29481632,28653153,32818441\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31476750> a 31476750 ;\n",
      "    ns9:abstract \" Investigate 3D (spatial and temporal) convolutional neural networks (CNNs) for real-time on-the-fly magnetic resonance imaging (MRI) reconstruction. In particular, we investigated the applicability of training CNNs on a patient-by-patient basis for the purpose of lung tumor segmentation. Data were acquired with our 3 T Philips Achieva system. A retrospective analysis was performed on six non-small cell lung cancer patients who received fully sampled dynamic acquisitions consisting of 650 free breathing images using a bSSFP sequence. We retrospectively undersampled the six patient's data by 5× and 10× acceleration. The retrospective data was used to quantitatively compare the CNN reconstruction to gold truth data via the Dice coefficient (DC) and centroid displacement to compare the tumor segmentations. Reconstruction noise was investigated using the normalized mean square error (NMSE). We further validated the technique using prospectively undersampled data from a volunteer and motion phantom. The retrospectively undersampled data at 5× and 10× acceleration was reconstructed using patient specific trained CNNs. The patient average DCs for the tumor segmentation at 5× and 10× acceleration were 0.94 and 0.92, respectively. These DC values are greater than the inter- and intra-observer segmentations acquired by radiation oncologist experts as reported in a previous study of ours. Furthermore, the patient specific CNN can be trained in under 6 h and the reconstruction time was 65 ms per image. The prospectively undersampled CNN reconstruction data yielded qualitatively acceptable images. We have shown that 3D CNNs can be used for real-time on-the-fly dynamic image reconstruction utilizing both spatial and temporal data in this proof of concept study. We evaluated the technique using six retrospectively undersampled lung cancer patient data sets, as well as prospectively undersampled data acquired from a volunteer and motion phantom. The reconstruction speed achieved for our current implementation was 65 ms per image. \"^^xsd:string ;\n",
      "    ns8:name \"Bryson Dietz; Jihyun Yun; Eugene Yip; Zsolt Gabos; B Gino Fallone; Keith Wachowicz; \"^^xsd:string ;\n",
      "    ns5:journal \"Phys Med Biol\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Humans;  Image Processing, Computer-Assisted / methods*;  Lung Neoplasms / diagnostic imaging*;  Lung Neoplasms / physiopathology;  Lung Neoplasms / radiotherapy*;  Magnetic Resonance Imaging*;  Movement;  Neural Networks, Computer*;  Respiration;  Time Factors;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Humans;  Image Processing, Computer-Assisted / methods*;  Lung Neoplasms / diagnostic imaging*;  Lung Neoplasms / physiopathology;  Lung Neoplasms / radiotherapy*;  Magnetic Resonance Imaging*;  Movement;  Neural Networks, Computer*;  Respiration;  Time Factors;  \"^^xsd:string ;\n",
      "    ns2:Title 31476750 ;\n",
      "    ns4:volume \"64\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://doi.org/10.1088/1361-6560/ab408e']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"32135531,28543069,28102958,30195984,32215571\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31603823> a 31603823 ;\n",
      "    ns9:abstract \" Studies using time-frequency analysis have reported that somatosensory evoked potentials provide information regarding the location of spinal cord injury. However, a better understanding of the time-frequency components derived from somatosensory evoked potentials is essential for developing more reliable algorithms that can diagnosis level (location) of cervical injury. In the present study, we proposed a random forests machine learning approach, for separating somatosensory evoked potentials depending on spinal cord state. For data acquisition, we established rat models of compression spinal cord injury at the C4, C5, and C6 levels to induce cervical myelopathy. After making the compression injury, we collected somatosensory evoked potentials and extracted their time-frequency components. We then used the random forests classification system to analyze the evoked potential dataset that was obtained from the three groups of model rats. Evaluation of the classifier performance revealed an overall classification accuracy of 84.72%, confirming that the random forests method was able to separate the time-frequency components of somatosensory evoked potentials from rats under different conditions. Features of the time-frequency components contained information that could identify the location of the cervical spinal cord injury, demonstrating the potential benefits of using time-frequency components of somatosensory evoked potentials to diagnose the level of cervical injury in cervical myelopathy. \"^^xsd:string ;\n",
      "    ns8:name \"Hongyan Cui; Yazhou Wang; Guangsheng Li; Yongcan Huang; Yong Hu; \"^^xsd:string ;\n",
      "    ns5:journal \"IEEE Trans Neural Syst Rehabil Eng\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Algorithms;  Animals;  Cervical Vertebrae / injuries*;  Evoked Potentials, Somatosensory*;  Machine Learning;  Rats;  Rats, Sprague-Dawley;  Spinal Cord Compression / diagnosis;  Spinal Cord Compression / physiopathology;  Spinal Cord Diseases / diagnosis*;  Spinal Cord Injuries / diagnosis*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Algorithms;  Animals;  Cervical Vertebrae / injuries*;  Evoked Potentials, Somatosensory*;  Machine Learning;  Rats;  Rats, Sprague-Dawley;  Spinal Cord Compression / diagnosis;  Spinal Cord Compression / physiopathology;  Spinal Cord Diseases / diagnosis*;  Spinal Cord Injuries / diagnosis*;  \"^^xsd:string ;\n",
      "    ns2:Title 31603823 ;\n",
      "    ns4:volume \"27\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/TNSRE.2019.2945634']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"28539587,25626775,11221784,23963005,18549397\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31658883> a 31658883 ;\n",
      "    ns9:abstract \" The purpose of this study was to develop an automated method for identifying and classifying change of direction (COD) movements in professional tennis using tracking data. Three sport science and strength and conditioning experts coded match-play footage of nineteen professional tennis players (9 male and 10 female) from the Australian Open Grand Slam for COD of medium and high intensity. A total of 1,494 changes were identified and aligned with 2D player position sampled at 25 Hz based on camera tracking data. Several machine learning classifiers were trained and tested on a set of 1,128 time-motion features. A random forest algorithm was found to have the best out-of-sample performance, classifying medium and high intensity changes with an F1-score of 0.729. This research offers a novel and applicable way for utilising player tracking data and machine learning techniques to automatically identify and classify COD movements in professional tennis. \"^^xsd:string ;\n",
      "    ns8:name \"Brandon Giles; Stephanie Kovalchik; Machar Reid; \"^^xsd:string ;\n",
      "    ns5:journal \"J Sports Sci\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Hawk-Eye; analytics; expert performance; tennis movement; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Hawk-Eye; analytics; expert performance; tennis movement;   Validation Study;  Adult;  Biomechanical Phenomena;  Competitive Behavior / classification;  Competitive Behavior / physiology;  Cross-Sectional Studies;  Female;  Humans;  Machine Learning*;  Male;  Motor Skills / classification*;  Motor Skills / physiology;  Reproducibility of Results;  Tennis / physiology*;  Time and Motion Studies;  Young Adult;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Validation Study;  Adult;  Biomechanical Phenomena;  Competitive Behavior / classification;  Competitive Behavior / physiology;  Cross-Sectional Studies;  Female;  Humans;  Machine Learning*;  Male;  Motor Skills / classification*;  Motor Skills / physiology;  Reproducibility of Results;  Tennis / physiology*;  Time and Motion Studies;  Young Adult;  \"^^xsd:string ;\n",
      "    ns2:Title 31658883 ;\n",
      "    ns4:volume \"38\"^^xsd:string ;\n",
      "    rdfs:comment \"['None']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29419342,28182523,30024896,7712556,28582628\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31718061> a 31718061 ;\n",
      "    ns9:abstract \" The work aiming to unravel the correlation between protein sequence and function in the absence of structural information can be highly rewarding. We present a new way of considering descriptors from the amino acids index database for modeling and predicting the fitness value of a polypeptide chain. This approach includes the following steps: (i) Calculating Q elementary numerical sequences (Ele_SEQ) depending on the encoding of the amino acid residues, (ii) determining an extended numerical sequence (Ext_SEQ) by concatenating the Q elementary numerical sequences, wherein at least one elementary numerical sequence is a protein spectrum obtained by applying fast Fourier transformation (FFT), and (iii) predicting a value of fitness for polypeptide variants (train and/or validation set). These new descriptors were tested on four sets of proteins of different lengths (GLP-2, TNF alpha, cytochrome P450, and epoxide hydrolase) and activities (cAMP activation, binding affinity, thermostability and enantioselectivity). We show that the use of multiple physicochemical descriptors coupled with the implementation of the FFT, taking into account the interactions between residues of amino amides within the protein sequence, could lead to very significant improvement in the quality of models and predictions. The choice of the descriptor or of the combination of descriptors and/or FFT is dependent on the couple protein/fitness. This approach can provide potential users with value added to existing mutant libraries where screening efforts have so far been unsuccessful in finding improved polypeptide mutants for useful applications. \"^^xsd:string ;\n",
      "    ns8:name \"Nicolas T Fontaine; Xavier F Cadet; Iyanar Vetrivel; \"^^xsd:string ;\n",
      "    ns5:journal \"Int J Mol Sci\"^^xsd:string ;\n",
      "    ns7:key_abs \"  artificial intelligence; digital signal processing; directed evolution; extended sequence; innov’SAR; machine learning; protein spectrum; rational screening; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  artificial intelligence; digital signal processing; directed evolution; extended sequence; innov’SAR; machine learning; protein spectrum; rational screening;   Animals;  Catalytic Domain;  Cytochrome P-450 Enzyme System / chemistry;  Cytochrome P-450 Enzyme System / metabolism;  Epoxide Hydrolases / chemistry;  Epoxide Hydrolases / metabolism;  Glucagon-Like Peptide-2 Receptor / chemistry;  Glucagon-Like Peptide-2 Receptor / metabolism;  Humans;  Machine Learning*;  Quantitative Structure-Activity Relationship*;  Sequence Analysis, Protein / methods*;  Tumor Necrosis Factor-alpha / chemistry;  Tumor Necrosis Factor-alpha / metabolism;  Glucagon-Like Peptide-2 Receptor;  Tumor Necrosis Factor-alpha;  Cytochrome P-450 Enzyme System;  Epoxide Hydrolases;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Animals;  Catalytic Domain;  Cytochrome P-450 Enzyme System / chemistry;  Cytochrome P-450 Enzyme System / metabolism;  Epoxide Hydrolases / chemistry;  Epoxide Hydrolases / metabolism;  Glucagon-Like Peptide-2 Receptor / chemistry;  Glucagon-Like Peptide-2 Receptor / metabolism;  Humans;  Machine Learning*;  Quantitative Structure-Activity Relationship*;  Sequence Analysis, Protein / methods*;  Tumor Necrosis Factor-alpha / chemistry;  Tumor Necrosis Factor-alpha / metabolism;  Glucagon-Like Peptide-2 Receptor;  Tumor Necrosis Factor-alpha;  Cytochrome P-450 Enzyme System;  Epoxide Hydrolases;  \"^^xsd:string ;\n",
      "    ns2:Title 31718061 ;\n",
      "    ns4:volume \"20\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.mdpi.com/resolver?pii=ijms20225640', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31718061/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30425279,30326841,29338232,17705863,19442071\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31752864> a 31752864 ;\n",
      "    ns9:abstract \"  Background:  Neuropsychological tests (NPTs) are important tools for informing diagnoses of cognitive impairment (CI). However, interpreting NPTs requires specialists and is thus time-consuming. To streamline the application of NPTs in clinical settings, we developed and evaluated the accuracy of a machine learning algorithm using multi-center NPT data.   Methods:  Multi-center data were obtained from 14,926 formal neuropsychological assessments (Seoul Neuropsychological Screening Battery), which were classified into normal cognition (NC), mild cognitive impairment (MCI) and Alzheimer's disease dementia (ADD). We trained a machine learning model with artificial neural network algorithm using TensorFlow (https://www.tensorflow.org) to distinguish cognitive state with the 46-variable data and measured prediction accuracies from 10 randomly selected datasets. The features of the NPT were listed in order of their contribution to the outcome using Recursive Feature Elimination.   Results:  The ten times mean accuracies of identifying CI (MCI and ADD) achieved by 96.66 ± 0.52% of the balanced dataset and 97.23 ± 0.32% of the clinic-based dataset, and the accuracies for predicting cognitive states (NC, MCI or ADD) were 95.49 ± 0.53 and 96.34 ± 1.03%. The sensitivity to the detection CI and MCI in the balanced dataset were 96.0 and 96.0%, and the specificity were 96.8 and 97.4%, respectively. The 'time orientation' and '3-word recall' score of MMSE were highly ranked features in predicting CI and cognitive state. The twelve features reduced from 46 variable of NPTs with age and education had contributed to more than 90% accuracy in predicting cognitive impairment.   Conclusions:  The machine learning algorithm for NPTs has suggested potential use as a reference in differentiating cognitive impairment in the clinical setting. \"^^xsd:string ;\n",
      "    ns8:name \"Min Ju Kang; Sang Yun Kim; Duk L Na; Byeong C Kim; Dong Won Yang; Eun-Joo Kim; Hae Ri Na; Hyun Jeong Han; Jae-Hong Lee; Jong Hun Kim; Kee Hyung Park; Kyung Won Park; Seol-Heui Han; Seong Yoon Kim; Soo Jin Yoon; Bora Yoon; Sang Won Seo; So Young Moon; YoungSoon Yang; Yong S Shim; Min Jae Baek; Jee Hyang Jeong; Seong Hye Choi; Young Chul Youn; \"^^xsd:string ;\n",
      "    ns5:journal \"BMC Med Inform Decis Mak\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Alzheimer’s disease; Dementia; Machine learning; Mild cognitive impairment; Neuropsychological test; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Alzheimer’s disease; Dementia; Machine learning; Mild cognitive impairment; Neuropsychological test;   Research Support, Non-U.S. Gov't;  Age Factors;  Algorithms;  Alzheimer Disease / diagnosis*;  Cognitive Dysfunction / diagnosis*;  Datasets as Topic;  Deep Learning;  Humans;  Machine Learning*;  Neural Networks, Computer;  Neuropsychological Tests*;  Sensitivity and Specificity;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Age Factors;  Algorithms;  Alzheimer Disease / diagnosis*;  Cognitive Dysfunction / diagnosis*;  Datasets as Topic;  Deep Learning;  Humans;  Machine Learning*;  Neural Networks, Computer;  Neuropsychological Tests*;  Sensitivity and Specificity;  \"^^xsd:string ;\n",
      "    ns2:Title 31752864 ;\n",
      "    ns4:volume \"19\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://bmcmedinformdecismak.biomedcentral.com/articles/10.1186/s12911-019-0974-x', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31752864/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30567554,29355115,29617684,24913723,29164602\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31805166> a 31805166 ;\n",
      "    ns9:abstract \" Coronary heart disease (CHD) is one of the leading causes of death worldwide; if suffering from CHD and being in its end-stage, the most advanced treatments are required, such as heart surgery and heart transplant. Moreover, it is not easy to diagnose CHD at the earlier stage; hospitals diagnose it based on various types of medical tests. Thus, by predicting high-risk people who are to suffer from CHD, it is significant to reduce the risks of developing CHD. In recent years, some research works have been done using data mining to predict the risk of developing diseases based on medical tests. In this study, we have proposed a reconstruction error (RE) based deep neural networks (DNNs); this approach uses a deep autoencoder (AE) model for estimating RE. Initially, a training dataset is divided into two groups by their RE divergence on the deep AE model that learned from the whole training dataset. Next, two DNN classifiers are trained on each group of datasets separately by combining a RE based new feature with other risk factors to predict the risk of developing CHD. For creating the new feature, we use deep AE model that trained on the only high-risk dataset. We have performed an experiment to prove how the components of our proposed method work together more efficiently. As a result of our experiment, the performance measurements include accuracy, precision, recall, F-measure, and AUC score reached 86.3371%, 91.3716%, 82.9024%, 86.9148%, and 86.6568%, respectively. These results show that the proposed AE-DNNs outperformed regular machine learning-based classifiers for CHD risk prediction. \"^^xsd:string ;\n",
      "    ns8:name \"Tsatsral Amarbayasgalan; Kwang Ho Park; Jong Yun Lee; Keun Ho Ryu; \"^^xsd:string ;\n",
      "    ns5:journal \"PLoS One\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Algorithms;  Coronary Disease / epidemiology*;  Coronary Disease / etiology*;  Deep Learning*;  Humans;  Neural Networks, Computer*;  Reproducibility of Results;  Risk Assessment;  Risk Factors;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Algorithms;  Coronary Disease / epidemiology*;  Coronary Disease / etiology*;  Deep Learning*;  Humans;  Neural Networks, Computer*;  Reproducibility of Results;  Risk Assessment;  Risk Factors;  \"^^xsd:string ;\n",
      "    ns2:Title 31805166 ;\n",
      "    ns4:volume \"14\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.plos.org/10.1371/journal.pone.0225991', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31805166/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29297288,29404850,31931701,28861708,27529225\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31838291> a 31838291 ;\n",
      "    ns9:abstract \"  Background:  Digital technologies, including robots, are being increasingly used in elderly care. Their impact on users carries implications for successfully integrating technological innovations into care. This study aims to identify the impacts of care-robot implementation on elderly-care service stakeholders.   Methods:  Impacts of care-robot implementation on users - care personnel and elderly clients - are identified from the data collected during a 10-week field study of the implementation process of the care robot 'Zora' in municipal elderly care services in Finland. The data were obtained from semi-participatory observation (27 sessions) of the robot engaging in rehabilitation efforts in two care homes and a geriatric rehabilitation hospital, and focus-group interviews conducted with 40 care workers and clients.   Results:  Robot use in elderly care is associated with multiple types of impacts with positive, negative, and neutral dimensions. These include impacts on interaction and activity for clients, and impacts on the work atmosphere, meaningfulness of work content, and professional development for care personnel. Impacts on personnel were related to the need for orientation, problems of time usage, and overall attitudes toward novelty and renewing of care service. The robot's presence stimulated the clients into exercising and interacting. The care workers perceived the clients' well-being both as a motivation to learn how to use robots as well as a justification for negative views.   Conclusions:  Care-robots like Zora have the potential for multi-faceted rehabilitative functions and can become part of care service with careful systemic planning with a specific focus on orientation. Many of the identified impacts were related to how the robot fits into the service processes. Distinguishing between positive, negative, or neutral dimensions of different impacts is important. \"^^xsd:string ;\n",
      "    ns8:name \"Helinä Melkas; Lea Hennala; Satu Pekkarinen; Ville Kyrki; \"^^xsd:string ;\n",
      "    ns5:journal \"Int J Med Inform\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Care robot; Elderly-care services; Finland; Human impact assessment; Implementation; ‘Zora’ robot; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Care robot; Elderly-care services; Finland; Human impact assessment; Implementation; ‘Zora’ robot;   Research Support, Non-U.S. Gov't;  Aged;  Delivery of Health Care / standards*;  Finland;  Health Personnel / psychology*;  Health Plan Implementation*;  Humans;  Motivation;  Patient Acceptance of Health Care / psychology*;  Robotics / instrumentation*;  Self-Help Devices / statistics & numerical data*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Aged;  Delivery of Health Care / standards*;  Finland;  Health Personnel / psychology*;  Health Plan Implementation*;  Humans;  Motivation;  Patient Acceptance of Health Care / psychology*;  Robotics / instrumentation*;  Self-Help Devices / statistics & numerical data*;  \"^^xsd:string ;\n",
      "    ns2:Title 31838291 ;\n",
      "    ns4:volume \"134\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S1386-5056(19)30049-8']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30354007,26131794,29575204,31348963,26132353,32471108\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31841865> a 31841865 ;\n",
      "    ns9:abstract \" Crash Detection is essential in providing timely information to traffic management centers and the public to reduce its adverse effects. Prediction of crash risk is vital for avoiding secondary crashes and safeguarding highway traffic. For many years, researchers have explored several techniques for early and precise detection of crashes to aid in traffic incident management. With recent advancements in data collection techniques, abundant real-time traffic data is available for use. Big data infrastructure and machine learning algorithms can utilize this data to provide suitable solutions for the highway traffic safety system. This paper explores the feasibility of using deep learning models to detect crash occurrence and predict crash risk. Volume, Speed and Sensor Occupancy data collected from roadside radar sensors along Interstate 235 in Des Moines, IA is used for this study. This real-world traffic data is used to design feature set for the deep learning models for crash detection and crash risk prediction. The results show that a deep model has better crash detection performance and similar crash prediction performance than state of the art shallow models. Additionally, a sensitivity analysis was conducted for crash risk prediction using data 1-minute, 5-minutes and 10-minutes prior to crash occurrence. It was observed that is hard to predict the crash risk of a traffic condition, 10 min prior to a crash. \"^^xsd:string ;\n",
      "    ns8:name \"Tingting Huang; Shuo Wang; Anuj Sharma; \"^^xsd:string ;\n",
      "    ns5:journal \"Accid Anal Prev\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Crash detection; Crash prediction; Deep learning; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Crash detection; Crash prediction; Deep learning;   Accidents, Traffic / prevention & control*;  Accidents, Traffic / statistics & numerical data;  Data Collection / methods*;  Deep Learning*;  Humans;  Logistic Models;  Risk Assessment / methods;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Accidents, Traffic / prevention & control*;  Accidents, Traffic / statistics & numerical data;  Data Collection / methods*;  Deep Learning*;  Humans;  Logistic Models;  Risk Assessment / methods;  \"^^xsd:string ;\n",
      "    ns2:Title 31841865 ;\n",
      "    ns4:volume \"135\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://linkinghub.elsevier.com/retrieve/pii/S0001-4575(19)30555-X']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"31935238,23932695,30390519,16625608,25841161\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31874640> a 31874640 ;\n",
      "    ns9:abstract \"  Background:  Tandem mass spectrometry allows biologists to identify and quantify protein samples in the form of digested peptide sequences. When performing peptide identification, spectral library search is more sensitive than traditional database search but is limited to peptides that have been previously identified. An accurate tandem mass spectrum prediction tool is thus crucial in expanding the peptide space and increasing the coverage of spectral library search.   Results:  We propose MS<sup>2</sup>CNN, a non-linear regression model based on deep convolutional neural networks, a deep learning algorithm. The features for our model are amino acid composition, predicted secondary structure, and physical-chemical features such as isoelectric point, aromaticity, helicity, hydrophobicity, and basicity. MS<sup>2</sup>CNN was trained with five-fold cross validation on a three-way data split on the large-scale human HCD MS<sup>2</sup> dataset of Orbitrap LC-MS/MS downloaded from the National Institute of Standards and Technology. It was then evaluated on a publicly available independent test dataset of human HeLa cell lysate from LC-MS experiments. On average, our model shows better cosine similarity and Pearson correlation coefficient (0.690 and 0.632) than MS<sup>2</sup>PIP (0.647 and 0.601) and is comparable with pDeep (0.692 and 0.642). Notably, for the more complex MS<sup>2</sup> spectra of 3+ peptides, MS<sup>2</sup>PIP is significantly better than both MS<sup>2</sup>PIP and pDeep.   Conclusions:  We showed that MS<sup>2</sup>CNN outperforms MS<sup>2</sup>PIP for 2+ and 3+ peptides and pDeep for 3+ peptides. This implies that MS<sup>2</sup>CNN, the proposed convolutional neural network model, generates highly accurate MS<sup>2</sup> spectra for LC-MS/MS experiments using Orbitrap machines, which can be of great help in protein and peptide identifications. The results suggest that incorporating more data for deep learning model may improve performance. \"^^xsd:string ;\n",
      "    ns8:name \"Yang-Ming Lin; Ching-Tai Chen; Jia-Ming Chang; \"^^xsd:string ;\n",
      "    ns5:journal \"BMC Genomics\"^^xsd:string ;\n",
      "    ns7:key_abs \"  Deep convolutional neural networks; Deep learning; Machine learning; Mass spectrum; Peptide; Protein identification; Spectral library search; Tandem mass spectrometry; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  Deep convolutional neural networks; Deep learning; Machine learning; Mass spectrum; Peptide; Protein identification; Spectral library search; Tandem mass spectrometry;   Evaluation Study;  Deep Learning;  HeLa Cells;  Humans;  Neural Networks, Computer*;  Peptides / chemistry;  Proteins / chemistry;  Sequence Analysis, Protein;  Tandem Mass Spectrometry / methods*;  Peptides;  Proteins;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Evaluation Study;  Deep Learning;  HeLa Cells;  Humans;  Neural Networks, Computer*;  Peptides / chemistry;  Proteins / chemistry;  Sequence Analysis, Protein;  Tandem Mass Spectrometry / methods*;  Peptides;  Proteins;  \"^^xsd:string ;\n",
      "    ns2:Title 31874640 ;\n",
      "    ns4:volume \"20\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6297-6', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31874640/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"29125736,31847456,23695934,22610447,30417778\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31905774> a 31905774 ;\n",
      "    ns9:abstract \" Enhancer-promoter interactions (EPIs) are crucial for transcriptional regulation. Mapping such interactions proves useful for understanding disease regulations and discovering risk genes in genome-wide association studies. Some previous studies showed that machine learning methods, as computational alternatives to costly experimental approaches, performed well in predicting EPIs from local sequence and/or local epigenomic data. In particular, deep learning methods were demonstrated to outperform traditional machine learning methods, and using DNA sequence data alone could perform either better than or almost as well as only utilizing epigenomic data. However, most, if not all, of these previous studies were based on randomly splitting enhancer-promoter pairs as training, tuning, and test data, which has recently been pointed out to be problematic; due to multiple and duplicating/overlapping enhancers (and promoters) in enhancer-promoter pairs in EPI data, such random splitting does not lead to independent training, tuning, and test data, thus resulting in model over-fitting and over-estimating predictive performance. Here, after correcting this design issue, we extensively studied the performance of various deep learning models with local sequence and epigenomic data around enhancer-promoter pairs. Our results confirmed much lower performance using either sequence or epigenomic data alone, or both, than reported previously. We also demonstrated that local epigenomic features were more informative than local sequence data. Our results were based on an extensive exploration of many convolutional neural network (CNN) and feed-forward neural network (FNN) structures, and of gradient boosting as a representative of traditional machine learning. \"^^xsd:string ;\n",
      "    ns8:name \"Mengli Xiao; Zhong Zhuang; Wei Pan; \"^^xsd:string ;\n",
      "    ns5:journal \"Genes (Basel)\"^^xsd:string ;\n",
      "    ns7:key_abs \"  boosting; convolutional neural networks; deep learning; feed-forward neural networks; machine learning; \"^^xsd:string ;\n",
      "    ns1:key_merged \"  boosting; convolutional neural networks; deep learning; feed-forward neural networks; machine learning;   Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Cell Line;  Computational Biology / methods*;  Deep Learning;  Enhancer Elements, Genetic;  Epigenomics / methods*;  Humans;  K562 Cells;  Neural Networks, Computer;  Promoter Regions, Genetic;  Whole Genome Sequencing / methods*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, N.I.H., Extramural;  Research Support, Non-U.S. Gov't;  Cell Line;  Computational Biology / methods*;  Deep Learning;  Enhancer Elements, Genetic;  Epigenomics / methods*;  Humans;  K562 Cells;  Neural Networks, Computer;  Promoter Regions, Genetic;  Whole Genome Sequencing / methods*;  \"^^xsd:string ;\n",
      "    ns2:Title 31905774 ;\n",
      "    ns4:volume \"11\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://www.mdpi.com/resolver?pii=genes11010041', 'https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31905774/']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30562350,30649185,31590652,25461775,27321818,32325634\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31946190> a 31946190 ;\n",
      "    ns9:abstract \" Electrical impedance tomography (EIT) is a noninvasive and non-radiative medical imaging technique based on detecting the inhomogeneous electrical properties of the tissue. The inverse problem of EIT is a highly nonlinear ill-posed problem, which is the main reason that affects image quality. Our goal is to solve the EIT inverse problem using the nonlinear mapping properties of artificial neural networks (ANNs) and convolutional neural networks (CNNs). In this paper, the adaptive moment estimation (ADAM) optimization method and mean-square-error (MSE) function are used to train an ANN to solve the inverse problem and a CNN to process the ANN image. The networks are trained on datasets of simulated data, and tested on datasets of simulated data and experimental data. Results for time-difference EIT (td-EIT) images are presented using simulated EIT data from EIDORS and experimental EIT data from our EIT systems. The results are used to compare the proposed method with the one-step Gauss-Newton linear method and RBFNN method. The proposed method offers improved resolution (RES), low position error (PE) and excellent artefact removal compared to the existing methods. The experimental results show that our method can improve the RES by 50 to 70 percent and reduce the PE by 60 to 70 percent. The improvements in RES and processing speed are essential for clinical EIT measurement of dynamic physiological processes. \"^^xsd:string ;\n",
      "    ns8:name \"Shu-Wei Huang; Hao-Min Cheng; Shien-Fong Lin; \"^^xsd:string ;\n",
      "    ns5:journal \"Conf Proc IEEE Eng Med Biol Soc\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Algorithms*;  Electric Impedance;  Humans;  Image Processing, Computer-Assisted*;  Neural Networks, Computer*;  Tomography*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Algorithms*;  Electric Impedance;  Humans;  Image Processing, Computer-Assisted*;  Neural Networks, Computer*;  Tomography*;  \"^^xsd:string ;\n",
      "    ns2:Title 31946190 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/EMBC.2019.8856781']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"12744177,29994023,29206856,12557783,18799834\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31946214> a 31946214 ;\n",
      "    ns9:abstract \" Safe interactions between humans and robots require the robotic arms and/or tools to recognize and react to the surrounding environment via pressure sensing. With small-scale surgical interventions in mind, we have developed a flexible skin with tens of pressure sensing elements, designed to cover a 5mm diameter tool. The prototype uses only biocompatible materials: soft silicones, carbon powder and metal wires. The material performance, sensing element design, manufacturing technology, and the readout electronics are described. Our prototype demonstrates the feasibility of using this technology in various intervention scenarios, from endoscopic navigation to tissue manipulation. We conclude by identifying research directions that maximise the potential of the proposed technology. \"^^xsd:string ;\n",
      "    ns8:name \"Piotr Wasylczyk; Filip Ozimek; Manish K Tiwari; Lyndon da Cruz; Christos Bergeles; \"^^xsd:string ;\n",
      "    ns5:journal \"Conf Proc IEEE Eng Med Biol Soc\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Research Support, Non-U.S. Gov't;  Electronics;  Equipment Design;  Humans;  Robotics;  Sensation*;  Skin*;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Research Support, Non-U.S. Gov't;  Electronics;  Equipment Design;  Humans;  Robotics;  Sensation*;  Skin*;  \"^^xsd:string ;\n",
      "    ns2:Title 31946214 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/EMBC.2019.8856871']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"26445045,31069938,26352901,27897993,26017446\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31946767> a 31946767 ;\n",
      "    ns9:abstract \" The use of ImageNet pre-trained networks is becoming widespread in the medical imaging community. It enables training on small datasets, commonly available in medical imaging tasks. The recent emergence of a large Chest X-ray dataset opened the possibility for learning features that are specific to the X-ray analysis task. In this work, we demonstrate that the features learned allow for better classification results for the problem of Tuberculosis detection and enable generalization to an unseen dataset.To accomplish the task of feature learning, we train a DenseNet-121 CNN on 112K images from the ChestXray14 dataset which includes labels of 14 common thoracic pathologies. In addition to the pathology labels, we incorporate meta-data which is available in the dataset: Patient Positioning, Gender and Patient Age. We term this architecture MetaChexNet. As a by-product of the feature learning, we demonstrate state of the art performance on the task of patient Age &amp; Gender estimation using CNN's. Finally, we show the features learned using ChestXray14 allow for better transfer learning on small-scale datasets for Tuberculosis. \"^^xsd:string ;\n",
      "    ns8:name \"Ophir Gozes; Hayit Greenspan; \"^^xsd:string ;\n",
      "    ns5:journal \"Conf Proc IEEE Eng Med Biol Soc\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Deep Learning*;  Humans;  Neural Networks, Computer*;  Radiography, Thoracic*;  Tuberculosis / diagnostic imaging*;  X-Rays;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Deep Learning*;  Humans;  Neural Networks, Computer*;  Radiography, Thoracic*;  Tuberculosis / diagnostic imaging*;  X-Rays;  \"^^xsd:string ;\n",
      "    ns2:Title 31946767 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/EMBC.2019.8856729']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"26886976,32524445,31715332,30802231,30730032,32834523\"^^xsd:string .\n",
      "\n",
      "<http://example.org/PMID/31946810> a 31946810 ;\n",
      "    ns9:abstract \" While cardiovascular diseases (CVDs) are commonly diagnosed by cardiologists via inspecting electrocardiogram (ECG) waveforms, these decisions can be supported by a data-driven approach, which may automate this process. An automatic diagnostic approach often employs hand-crafted features extracted from ECG waveforms. These features, however, do not generalise well, challenged by variation in acquisition settings such as sampling rate and mounting points. Existing deep learning (DL) approaches, on the other hand, extract features from ECG automatically but require construction of dedicated networks that require huge data and computational resource if trained from scratch. Here we propose an end-to-end trainable cross-domain transfer learning for CVD classification from ECG waveforms, by utilising existing vision-based CNN frameworks as feature extractors, followed by ECG feature learning layers. Because these frameworks are designed for image inputs, we employ a stacked spectrogram representation of multi-lead ECG waveforms as a preprocessing step. We also proposed a fusion of multiple ECG leads, using plausible stacking arrangements of the spectrograms, to encode their spatial relations. The proposed approach is validated on multiple ECG datasets and competitive performance is achieved. \"^^xsd:string ;\n",
      "    ns8:name \"Girmaw Abebe Tadesse; Tingting Zhu; Yong Liu; Yingling Zhou; Jiyan Chen; Maoyi Tian; David Clifton; \"^^xsd:string ;\n",
      "    ns5:journal \"Conf Proc IEEE Eng Med Biol Soc\"^^xsd:string ;\n",
      "    ns7:key_abs \"\"^^xsd:string ;\n",
      "    ns1:key_merged \"  Cardiovascular Diseases / diagnosis*;  Deep Learning*;  Diagnosis, Computer-Assisted*;  Electrocardiography*;  Humans;  \"^^xsd:string ;\n",
      "    ns6:key_mesh \"Cardiovascular Diseases / diagnosis*;  Deep Learning*;  Diagnosis, Computer-Assisted*;  Electrocardiography*;  Humans;  \"^^xsd:string ;\n",
      "    ns2:Title 31946810 ;\n",
      "    ns4:volume \"2019\"^^xsd:string ;\n",
      "    rdfs:comment \"['https://dx.doi.org/10.1109/EMBC.2019.8857737']\"^^xsd:string ;\n",
      "    rdfs:seeAlso \"30245122,30524025,31330925,31153403,31888592\"^^xsd:string .\n",
      "\n",
      "ns3:24694170 ns3:doi \"10.1142/S0129065714500154\"^^xsd:integer .\n",
      "\n",
      "ns3:24955405 ns3:doi \"10.1155/2014/468176\"^^xsd:integer .\n",
      "\n",
      "ns3:25291742 ns3:doi \"10.1109/TNNLS.2014.2297991\"^^xsd:integer .\n",
      "\n",
      "ns3:25570647 ns3:doi \"10.1109/EMBC.2014.6944279\"^^xsd:integer .\n",
      "\n",
      "ns3:28820010 ns3:doi \"10.1177/1460458217723169\"^^xsd:integer .\n",
      "\n",
      "ns3:29237568 ns3:doi \"10.1016/j.neuroscience.2017.11.055\"^^xsd:integer .\n",
      "\n",
      "ns3:29546796 ns3:doi \"10.1080/07391102.2018.1453377\"^^xsd:integer .\n",
      "\n",
      "ns3:29990046 ns3:doi \"10.1109/TCBB.2017.2784434\"^^xsd:integer .\n",
      "\n",
      "ns3:29993442 ns3:doi \"10.1109/TBME.2018.2830366\"^^xsd:integer .\n",
      "\n",
      "ns3:29994031 ns3:doi \"10.1109/TCBB.2018.2831666\"^^xsd:integer .\n",
      "\n",
      "ns3:30040632 ns3:doi \"10.1109/TMI.2018.2857800\"^^xsd:integer .\n",
      "\n",
      "ns3:30123948 ns3:doi \"10.1007/s11517-018-1876-2\"^^xsd:integer .\n",
      "\n",
      "ns3:30189255 ns3:doi \"10.1016/j.jbi.2018.09.001\"^^xsd:integer .\n",
      "\n",
      "ns3:30206809 ns3:doi \"10.1007/s11538-018-0493-4\"^^xsd:integer .\n",
      "\n",
      "ns3:30334752 ns3:doi \"10.1109/TMI.2018.2875868\"^^xsd:integer .\n",
      "\n",
      "ns3:30346279 ns3:doi \"10.1109/TBME.2018.2877577\"^^xsd:integer .\n",
      "\n",
      "ns3:30522862 ns3:doi \"10.1016/j.tibs.2018.10.010\"^^xsd:integer .\n",
      "\n",
      "ns3:30526357 ns3:doi \"10.1148/radiol.2018182573\"^^xsd:integer .\n",
      "\n",
      "ns3:30528509 ns3:doi \"10.1016/j.compbiolchem.2018.11.003\"^^xsd:integer .\n",
      "\n",
      "ns3:30575527 ns3:doi \"10.1109/TBME.2018.2888998\"^^xsd:integer .\n",
      "\n",
      "ns3:30613101 ns3:doi \"10.1038/s41591-018-0318-x\"^^xsd:integer .\n",
      "\n",
      "ns3:30617720 ns3:doi \"10.1007/s11548-018-01908-8\"^^xsd:integer .\n",
      "\n",
      "ns3:30624652 ns3:doi \"10.1093/database/bay138\"^^xsd:integer .\n",
      "\n",
      "ns3:30632459 ns3:doi \"10.1080/09593330.2019.1567609\"^^xsd:integer .\n",
      "\n",
      "ns3:30653830 ns3:doi \"10.1002/ps.5337\"^^xsd:integer .\n",
      "\n",
      "ns3:30670255 ns3:doi \"10.1016/j.ctim.2018.12.001\"^^xsd:integer .\n",
      "\n",
      "ns3:30678171 ns3:doi \"10.3390/molecules24030380\"^^xsd:integer .\n",
      "\n",
      "ns3:30682710 ns3:doi \"10.1016/j.neunet.2018.12.002\"^^xsd:integer .\n",
      "\n",
      "ns3:30702339 ns3:doi \"10.1123/ijspp.2017-0864\"^^xsd:integer .\n",
      "\n",
      "ns3:30712601 ns3:doi \"10.1016/j.cmpb.2018.12.032\"^^xsd:integer .\n",
      "\n",
      "ns3:30716040 ns3:doi \"10.1109/TNSRE.2019.2896659\"^^xsd:integer .\n",
      "\n",
      "ns3:30753964 ns3:doi \"10.1016/j.neunet.2019.01.009\"^^xsd:integer .\n",
      "\n",
      "ns3:30794638 ns3:doi \"10.1371/journal.pone.0212665\"^^xsd:integer .\n",
      "\n",
      "ns3:30817960 ns3:doi \"10.1016/j.urology.2019.02.017\"^^xsd:integer .\n",
      "\n",
      "ns3:30834812 ns3:doi \"10.1177/1539449218825438\"^^xsd:integer .\n",
      "\n",
      "ns3:30840663 ns3:doi \"10.1371/journal.pone.0212935\"^^xsd:integer .\n",
      "\n",
      "ns3:30852421 ns3:doi \"10.1016/j.ejca.2019.02.005\"^^xsd:integer .\n",
      "\n",
      "ns3:30861015 ns3:doi \"10.1371/journal.pone.0212844\"^^xsd:integer .\n",
      "\n",
      "ns3:30876845 ns3:doi \"10.1016/j.drudis.2019.03.003\"^^xsd:integer .\n",
      "\n",
      "ns3:30941889 ns3:doi \"10.1002/pmic.201900019\"^^xsd:integer .\n",
      "\n",
      "ns3:31022847 ns3:doi \"10.3390/s19081923\"^^xsd:integer .\n",
      "\n",
      "ns3:31029054 ns3:doi \"10.1016/j.neunet.2019.04.010\"^^xsd:integer .\n",
      "\n",
      "ns3:31029061 ns3:doi \"10.1093/gigascience/giz044\"^^xsd:integer .\n",
      "\n",
      "ns3:31039552 ns3:doi \"10.1088/1741-2552/ab1e01\"^^xsd:integer .\n",
      "\n",
      "ns3:31094833 ns3:doi \"10.1097/CORR.0000000000000687\"^^xsd:integer .\n",
      "\n",
      "ns3:31122081 ns3:doi \"10.1080/09546634.2019.1623373\"^^xsd:integer .\n",
      "\n",
      "ns3:31135114 ns3:doi \"10.12116/j.issn.1004-5619.2019.02.012\"^^xsd:integer .\n",
      "\n",
      "ns3:31136171 ns3:doi \"10.1021/acs.jcim.9b00264\"^^xsd:integer .\n",
      "\n",
      "ns3:31145684 ns3:doi \"10.1515/bmt-2018-0012\"^^xsd:integer .\n",
      "\n",
      "ns3:31278668 ns3:doi \"10.1007/978-1-4939-9074-0_9\"^^xsd:integer .\n",
      "\n",
      "ns3:31285275 ns3:doi \"10.1523/ENEURO.0362-18.2019\"^^xsd:integer .\n",
      "\n",
      "ns3:31317398 ns3:doi \"10.1007/s12149-019-01386-1\"^^xsd:integer .\n",
      "\n",
      "ns3:31320356 ns3:doi \"10.1136/bmjopen-2019-029621\"^^xsd:integer .\n",
      "\n",
      "ns3:31333196 ns3:doi \"10.2196/13809\"^^xsd:integer .\n",
      "\n",
      "ns3:31374758 ns3:doi \"10.1109/ICORR.2019.8779534\"^^xsd:integer .\n",
      "\n",
      "ns3:31374796 ns3:doi \"10.1109/ICORR.2019.8779461\"^^xsd:integer .\n",
      "\n",
      "ns3:31399493 ns3:doi \"10.1523/ENEURO.0201-19.2019\"^^xsd:integer .\n",
      "\n",
      "ns3:31400112 ns3:doi \"10.1093/nar/gkz654\"^^xsd:integer .\n",
      "\n",
      "ns3:31416053 ns3:doi \"10.1088/1361-6560/ab3b86\"^^xsd:integer .\n",
      "\n",
      "ns3:31416552 ns3:doi \"10.1016/j.cmpb.2019.06.030\"^^xsd:integer .\n",
      "\n",
      "ns3:31422557 ns3:doi \"10.1007/s11517-019-02027-5\"^^xsd:integer .\n",
      "\n",
      "ns3:31437939 ns3:doi \"10.3233/SHTI190237\"^^xsd:integer .\n",
      "\n",
      "ns3:31455677 ns3:doi \"10.1534/g3.119.400596\"^^xsd:integer .\n",
      "\n",
      "ns3:31466478 ns3:doi \"10.2217/epi-2019-0206\"^^xsd:integer .\n",
      "\n",
      "ns3:31470662 ns3:doi \"10.3390/cells8091004\"^^xsd:integer .\n",
      "\n",
      "ns3:31476750 ns3:doi \"10.1088/1361-6560/ab408e\"^^xsd:integer .\n",
      "\n",
      "ns3:31603823 ns3:doi \"10.1109/TNSRE.2019.2945634\"^^xsd:integer .\n",
      "\n",
      "ns3:31658883 ns3:doi \"10.1080/02640414.2019.1684132\"^^xsd:integer .\n",
      "\n",
      "ns3:31718061 ns3:doi \"10.3390/ijms20225640\"^^xsd:integer .\n",
      "\n",
      "ns3:31752864 ns3:doi \"10.1186/s12911-019-0974-x\"^^xsd:integer .\n",
      "\n",
      "ns3:31805166 ns3:doi \"10.1371/journal.pone.0225991\"^^xsd:integer .\n",
      "\n",
      "ns3:31838291 ns3:doi \"10.1016/j.ijmedinf.2019.104041\"^^xsd:integer .\n",
      "\n",
      "ns3:31841865 ns3:doi \"10.1016/j.aap.2019.105392\"^^xsd:integer .\n",
      "\n",
      "ns3:31874640 ns3:doi \"10.1186/s12864-019-6297-6\"^^xsd:integer .\n",
      "\n",
      "ns3:31905774 ns3:doi \"10.3390/genes11010041\"^^xsd:integer .\n",
      "\n",
      "ns3:31946190 ns3:doi \"10.1109/EMBC.2019.8856781\"^^xsd:integer .\n",
      "\n",
      "ns3:31946214 ns3:doi \"10.1109/EMBC.2019.8856871\"^^xsd:integer .\n",
      "\n",
      "ns3:31946767 ns3:doi \"10.1109/EMBC.2019.8856729\"^^xsd:integer .\n",
      "\n",
      "ns3:31946810 ns3:doi \"10.1109/EMBC.2019.8857737\"^^xsd:integer .\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(g.serialize(format='turtle').decode('UTF-8'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "q = \"select ?title ?abstract where {?title ?abstract <http://example.org/PMID/24694170>}\"\n",
    "x = g.query(q)\n",
    "t = list(x)\n",
    "print(\"----------------------------------------------------------------------------------------------\")\n",
    "for y in range(0,len(t)):\n",
    "    ss=t[y][0]\n",
    "    #search for id\n",
    "    q = \"select ?relation ?part  where {<\"+ss+\"> ?relation ?part}\"\n",
    "    xx = g.query(q)\n",
    "    print(xx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
